---
title: |
  | Longitudinal Transcriptomics analysis of CCHF infected patients
author: | 
  | Anoop Ambikan
  | Division of Clinical Microbiology
output:
  rmarkdown::pdf_document:
    fig_caption: true
    toc_depth: 3
    number_sections: true
    includes:  
      in_header: new_header.tex
bibliography: /home/anoop/Desktop/Anoop/Tutorial/Mybibliography.bib
csl: biomed-central.csl
sansfont: Arial
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

# \textcolor{blue}{Project Description}

Transcriptomics data was generated from peripheral blood obtained from three time points of CCHF infection. Data was generated for samples taken at day of hospitalization (Acute), day of discharge and at one month after discharge. Al the raw fastq files generated were processed using the rnaseq (doi: 10.5281/zenodo.1400710) pipeline from nf-core [@nfcore] community. The pipeline uses Trim Galore (https://github.com/FelixKrueger/TrimGalore) [@cutadapt] for adapter trimming with default options. Human reference genome version GRCh38 obtained from Ensembl was  used for alignment and read count estimation using aligner option star_salmon provided by the pipeline. Gene level estimated count and TPM of genes balong to biotypes namely, protein-coding, lincRNAs, IG gene and TR gene (https://www.ensembl.org/info/genome/genebuild/biotypes.html) were used for all the downstream analysis. The analysis pipeline executes the following tasks,

&nbsp;&nbsp;&nbsp;&nbsp;

&nbsp;&nbsp;&nbsp;&nbsp;**1. Sample distribution**

&nbsp;&nbsp;&nbsp;&nbsp;**2. Digital cell quantification**

&nbsp;&nbsp;&nbsp;&nbsp;**3. Differential gene expression analysis**

&nbsp;&nbsp;&nbsp;&nbsp;**5. Gene set enrichment analysis**


# \textcolor{blue}{Sample distribution}

Similarity and dissimilarity between the samples are examined by visualizing them in 2-Dimensional space. Dimensionality of the data was first reduced using Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP) using the R package umap v0.2.7.0. TPM normalized gene expression table was used for UMAP dimensionality reduction and the result is visualized in Figure 1.

```{r,eval=FALSE}
library("tximport")
library("tximportData")
design=read.delim("/home/anoop/Desktop/CCHF/RNASeq/Re_Seq/Sample_Info.txt",header = TRUE)
files <- file.path("/home/anoop/Desktop/CCHF/RNASeq/Re_Seq",design$Sample_ID, "CodingCount.txt")
names(files) <- design$Sample_ID
file.exists(files)
txi <- tximport(files, type="salmon",txIn=FALSE,geneIdCol="Name")
write.table(txi$abundance,file = "/home/anoop/Desktop/CCHF/RNASeq/Re_Seq/TPM_tmp.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r,eval=FALSE}
library(umap)
data=read.delim("/home/anoop/Desktop/CCHF/RNASeq/Re_Seq/TPM.txt",header=TRUE,row.names = 1,check.names = FALSE)
Art.umap = umap(t(data))
write.table(Art.umap$layout,file="/home/anoop/Desktop/CCHF/RNASeq/Re_Seq/TPM_UMAP.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/RNASeq/Re_Seq/TPM_UMAP.txt",row.names = 1,check.names = FALSE)
head(dat)
pdf("/home/anoop/Desktop/CCHF/RNASeq/Re_Seq/TPM_UMAP.pdf")
ggplot(dat, aes(x=V1, y=V2,color=`Time of Infection`)) + geom_point(size=3.5,aes(fill=`Time of Infection`),pch=21)+theme_gray()+
    scale_color_manual(values=c("Healthy"="#315b7d","Day of Hospitalization"="#740000",
                                "Day of Discharge"="#ca6666","One month after discharge"="#486129"))+
    scale_fill_manual(values=c("Healthy"="#4682B4","Day of Hospitalization"="#a70000",
                               "Day of Discharge"="#e4b2b2","One month after discharge"="#738957"))+
  labs(x="UMAP1",y="UMAP2")+
  theme(axis.title = element_text(size=10),legend.position = "bottom",plot.margin = margin(2,2,2,2, "cm"),
        legend.title=element_text(size=10),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  guides(shape = guide_legend(nrow =2,override.aes=list(fill="grey",color="grey")),color = guide_legend(nrow =2),fill = guide_legend(ncol =1))
dev.off()
```

```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/RNASeq/Re_Seq/TPM_UMAP.txt",row.names = 1,check.names = FALSE)
head(dat)
pdf("/home/anoop/Desktop/CCHF/RNASeq/Re_Seq/TPM_UMAP_SGS.pdf")
ggplot(dat, aes(x=V1, y=V2,color=`SGS category`)) + geom_point(size=3.5,aes(fill=`SGS category`),pch=21)+theme_gray()+
  labs(x="UMAP1",y="UMAP2")+
  theme(axis.title = element_text(size=10),legend.position = "bottom",plot.margin = margin(2,2,2,2, "cm"),
        legend.title=element_text(size=10),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  guides(shape = guide_legend(nrow =2,override.aes=list(fill="grey",color="grey")),color = guide_legend(nrow =2),fill = guide_legend(ncol =1))
dev.off()
```

```{r,eval=FALSE}
library(PCAtools)
count=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/TPM.txt",row.names = 1,check.names = FALSE)
meta=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Sample_Info.txt",row.names = 1)
p <- PCAtools::pca((count), metadata = meta, removeVar = 0.1)
write.table(p$rotated,file = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/TPM_PCA.txt",sep = "\t",col.names = NA,quote = FALSE)
head(p$variance)

```

```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/TPM_PCA.txt",row.names = 1,check.names = FALSE)
gg <- merge(dat,aggregate(cbind(mean.pc1=PC1,mean.pc2=PC2)~Cohort,dat,mean),by="Cohort")
head(gg)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/TPM_PCA.pdf")
ggplot(gg, aes(PC1,PC2,color=Cohort))+geom_point(size=3,aes(fill=Cohort,shape=SGS2))+
  scale_shape_manual(values=c("Healthy"=21, "Mild"=22, "Moderate"=23,"Severe"=24))+
  geom_point(aes(x=mean.pc1,y=mean.pc2,fill=Cohort,color=Cohort),size=3.5)+theme_gray()+
  geom_segment(aes(x=mean.pc1, y=mean.pc2, xend=PC1, yend=PC2))+
    scale_color_manual(values=c("Healthy"="#486129","T0"="#740000","T1"="#ca6666","T2"="#315b7d"))+
    scale_fill_manual(values=c("Healthy"="#738957","T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"))+
  labs(x="PC1, 98.44% variation",y="PC2, 0.51% variation")+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = Cohort), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```

```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/TPM_PCA.txt",row.names = 1,check.names = FALSE)
gg <- merge(dat,aggregate(cbind(mean.pc1=PC1,mean.pc2=PC2)~Cohort,dat,mean),by="Cohort")
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/TPM_PCA2.pdf")
ggplot(gg, aes(PC1,PC2,color=SGS))+geom_point(size=3,aes(fill=SGS),shape=21)+
  geom_point(aes(x=mean.pc1,y=mean.pc2,fill=SGS,color=SGS),size=3.5)+theme_gray()+
  geom_segment(aes(x=mean.pc1, y=mean.pc2, xend=PC1, yend=PC2))+
  scale_color_manual(values=c("Healthy"="#486129","T0_Mild"="#9d4c4c","T0_Moderate"="#740000","T0_Severe"="#510000",
                                                "T1_Mild"="#d99393","T1_Moderate"="#ca6666","T1_Severe"="#a15151",
                                                "T2_Mild"="#6e8ca4","T2_Moderate"="#315b7d","T2_Severe"="#223f57"))+
  scale_fill_manual(values=c("Healthy"="#486129","T0_Mild"="#9d4c4c","T0_Moderate"="#740000","T0_Severe"="#510000",
                                                "T1_Mild"="#d99393","T1_Moderate"="#ca6666","T1_Severe"="#a15151",
                                                "T2_Mild"="#6e8ca4","T2_Moderate"="#315b7d","T2_Severe"="#223f57"))+
  labs(x="PC1, 98.44% variation",y="PC2, 0.51% variation")+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```


```{r,eval=FALSE}
library(EPIC)
bulk <- read.delim(file="/home/anoop/Desktop/CCHF_NEW/CCHF/RNASeq/Re_Seq/EPIC/TPM_GeneNames.txt", row.names = 1)
bulk=as.matrix(bulk)
REF <- read.delim(file="/home/anoop/Desktop/CCHF_NEW/Blood_Cells_casted.txt", row.names = 1)
REF=as.matrix(REF)
sig=read.delim("/home/anoop/Desktop/CCHF_NEW/NewSigGenes.txt",header = FALSE)
SIG <- as.vector(sig[,1])

refList <- list(refProfiles=REF,sigGenes=SIG)
Res2 <- EPIC(bulk = bulk, reference = refList,scaleExprs=FALSE)

write.table(Res2$mRNAProportions,file="/home/anoop/Desktop/CCHF_NEW/CCHF/RNASeq/Re_Seq/EPIC/mRNAProp.txt",col.names = NA,quote = FALSE,sep="\t")
write.table(Res2$cellFractions,file="/home/anoop/Desktop/CCHF_NEW/CCHF/RNASeq/Re_Seq/EPIC/CellFrac.txt",col.names = NA,quote = FALSE,sep="\t")
write.table(Res2$fit.gof,file="/home/anoop/Desktop/CCHF_NEW/CCHF/RNASeq/Re_Seq/EPIC/OtherRes.txt",col.names = NA,quote = FALSE,sep="\t")

```

```{r,fig.align = "center",fig.height=13, fig.width=10, echo=FALSE,fig.cap ="Cell type proportions computed. Pvalues corresponding to Mann-Whiteny test are labelled",warning=FALSE,message=FALSE}
Data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/EPIC/figures/TOI.txt",header = TRUE,check.names = FALSE)
library(reshape2)
M=melt(Data)
library(ggpubr)
library(ggplot2)
my_comparisons = list( c("Healthy", "T0"),c("Healthy", "T1"),c("Healthy", "T2"),
                       c("T0", "T1"),c("T0", "T2"),
                       c("T1", "T2"))
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/EPIC/figures/EPIC.pdf",width = 15,height = 12)
ggplot(M,aes(x=factor(Time_of_Infection,levels = unique(Time_of_Infection)),y=value,fill=Time_of_Infection,color=Time_of_Infection))+geom_boxplot(outlier.shape = NA)+
  facet_wrap(~ variable, ncol = 4,scales ="free")+
  geom_jitter(shape=1, size=2.5,color="black",position=position_jitter(0.05))+
scale_color_manual(values=c("T2"="#315b7d","T0"="#740000",
                                "T1"="#ca6666","Healthy"="#486129"))+
    scale_fill_manual(values=c("T2"="#4682B4","T0"="#a70000",
                               "T1"="#e4b2b2","Healthy"="#738957"))+
  theme_bw()+stat_compare_means(label = "p.format",comparisons = my_comparisons,size=3)+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank(),legend.position ="bottom",panel.border = element_blank(),
        axis.ticks.x = element_blank(),axis.text.x = element_blank(),legend.title = element_blank(),legend.text = element_text(size = 15),
        plot.margin = margin(0.5,6,0.5,6, "cm"),strip.text.x = element_text(size = 13),
        axis.text.y = element_text(color="black",size=12),plot.title = element_text(hjust = 0.5,size = 15))+
  guides(color = guide_legend(nrow = 1),fill = guide_legend(nrow = 1))
dev.off()
```


```{r,fig.align = "center",fig.height=13, fig.width=10, echo=FALSE,fig.cap ="Cell type proportions computed. Pvalues corresponding to Mann-Whiteny test are labelled",warning=FALSE,message=FALSE}

iqr = function(z, lower = 0.25, upper = 0.75) {
  data.frame(
    y = median(z),
    ymin = quantile(z, lower),
    ymax = quantile(z, upper)
  )
}

Data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/EPIC/figures/TOI.txt",header = TRUE,check.names = FALSE)
library(reshape2)
M=melt(Data)
library(ggpubr)
library(ggplot2)
my_comparisons = list( c("Healthy", "T0"),c("Healthy", "T1"),c("Healthy", "T2"),
                       c("T0", "T1"),c("T0", "T2"),
                       c("T1", "T2"))
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/EPIC/figures/EPIC.pdf",width = 15,height = 12)
ggplot(M,aes(x=factor(Time_of_Infection,levels = unique(Time_of_Infection)),y=value))+
  facet_wrap(~ variable, ncol = 4,scales ="free")+
  geom_jitter(data = M,shape=21, size=3.5,aes(color=Time_of_Infection,fill=Time_of_Infection),position=position_jitter(0.1))+
 stat_summary(fun.data=iqr,
        geom="errorbar", color="black", width=0.2) +
  stat_summary(fun.y=median, geom="point", color="black",size=3)+
    scale_color_manual(values=c("Healthy"="#315b7d","T0"="#740000",
                                "T1"="#ca6666","T2"="#486129"))+
    scale_fill_manual(values=c("Healthy"="#4682B4","T0"="#a70000",
                               "T1"="#e4b2b2","T2"="#738957"))+
  theme_bw()+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank(),legend.position ="bottom",panel.border = element_blank(),
        axis.ticks.x = element_blank(),axis.text.x = element_blank(),legend.title = element_blank(),legend.text = element_text(size = 15),
        plot.margin = margin(0.5,3,0.5,3, "cm"),strip.text.x = element_text(size = 13),
        axis.text.y = element_text(color="black",size=12),plot.title = element_text(hjust = 0.5,size = 15))+
  guides(color = guide_legend(nrow = 1),fill = guide_legend(nrow = 1))
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/EPIC/figures/AvgPerc.txt")
M=reshape2::melt(data)
head(M)
#scale_color_manual(values=c(HC="#48690E",Mild="#ffd700",Severe="#ffa500"))+
library(ggplot2)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/EPIC/figures/AvgPerc.pdf")
ggplot(M) +
  geom_point(data=M,aes(x=variable,y=factor(CellType,levels = unique(CellType)),size=value,color=value))+theme_bw()+
  scale_color_gradient(low="#669cc0",high="#00365a")+
  theme(axis.title.y = element_blank(),axis.title.x = element_blank(),axis.text.x =element_text(size=10,color="black",angle =45, vjust =1.2, hjust=1),
        axis.text.y = element_text(size=10,color="black"),plot.title = element_text(hjust = 0.5),plot.margin = margin(5,3.5,5,3.5, "cm"),
        axis.ticks = element_blank(),legend.text = element_text(colour = "black"),
        legend.direction = "vertical",
        legend.title = element_text(size=10),panel.border = element_blank(),panel.grid.major = element_blank())+
  guides(size=guide_legend(override.aes=list(colour="grey"),title = "Average Cell Percentage",ncol = 1),color=guide_legend(title = NULL))+
  scale_size_continuous(range = c(2,5))
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/EPIC/figures/MedianPerc.txt")
M=reshape2::melt(data)
head(M)
#scale_color_manual(values=c(HC="#48690E",Mild="#ffd700",Severe="#ffa500"))+
library(ggplot2)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/EPIC/figures/MedianPerc.pdf")
ggplot(M) +
  geom_point(data=M,aes(x=variable,y=factor(CellType,levels = unique(CellType)),size=value,color=value))+theme_bw()+
  scale_color_gradient(low="#ffb732",high="#b27300")+
  theme(axis.title.y = element_blank(),axis.title.x = element_blank(),axis.text.x =element_text(size=10,color="black",angle =45, vjust =1.2, hjust=1),
        axis.text.y = element_text(size=10,color="black"),plot.title = element_text(hjust = 0.5),plot.margin = margin(5,3.5,5,3.5, "cm"),
        axis.ticks = element_blank(),legend.text = element_text(colour = "black"),
        legend.direction = "vertical",
        legend.title = element_text(size=10),panel.border = element_blank(),panel.grid.major = element_blank())+
  guides(size=guide_legend(override.aes=list(colour="grey"),title = "Median Cell Percentage",ncol = 1),color=guide_legend(title = NULL))+
  scale_size_continuous(range = c(2,5))
dev.off()
```


```{r,eval=FALSE}

## Healthy vs Day of Hospitalization

library(tximport)
library(tximportData)
library(DESeq2)
library(RUVSeq)
library(dplyr)

design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Design.txt",header = TRUE)
files <- file.path("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq",design$ID, "CodingCount.txt")
names(files) <- design$ID
txi <- tximport(files, type="salmon",txIn=FALSE,geneIdCol="Name")

count=as.data.frame(txi$counts)
count=round(count, 0)

Design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Design.txt",row.names = 1, check.names = FALSE)
filtered=count%>%filter(rowSums(across(where(is.numeric)))!=0)
Myset <- newSeqExpressionSet(as.matrix(filtered),phenoData = data.frame(Design))
Myset <- betweenLaneNormalization(Myset, which="upper")
design <- model.matrix(~TOI+Age+Gender+intermediate_monocyte+plasmacytoid_DC+non_classical_monocyte+neutrophil+otherCells+memory_CD4_Tcell+Nkcell, data=pData(Myset))
genes <- rownames(filtered)[grep("^ENS", rownames(filtered))]

y <- DGEList(counts=counts(Myset), group=as.factor(Design$TOI))
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
res <- residuals(fit, type="deviance")
Myset2 <- RUVr(Myset, genes, k=1, res)

ddsTxi <- DESeqDataSetFromTximport(txi,colData = pData(Myset2),design = ~ W_1 + TOI)
dds=DESeq(ddsTxi)
res=results(dds,contrast = c("TOI","Day_of_Hospitalization","Healthy"),independentFiltering = FALSE)                       
write.table(as.data.frame(res),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Result_RUV.txt",sep="\t", quote=FALSE,col.names = NA)

```


```{r,eval=FALSE}

## Healthy vs Day of Discharge

library(tximport)
library(tximportData)
library(DESeq2)
library(RUVSeq)
library(dplyr)

design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Design.txt",header = TRUE)
files <- file.path("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq",design$ID, "CodingCount.txt")
names(files) <- design$ID
txi <- tximport(files, type="salmon",txIn=FALSE,geneIdCol="Name")

count=as.data.frame(txi$counts)
count=round(count, 0)

Design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Design.txt",row.names = 1, check.names = FALSE)
filtered=count%>%filter(rowSums(across(where(is.numeric)))!=0)
Myset <- newSeqExpressionSet(as.matrix(filtered),phenoData = data.frame(Design))
Myset <- betweenLaneNormalization(Myset, which="upper")
design <- model.matrix(~TOI+Age+Gender+intermediate_monocyte+memory_Bcell+memory_CD4_Tcell+plasmacytoid_DC+MAIT_Tcell+otherCells+myeloid_DC+Nkcell+classical_monocyte, data=pData(Myset))
genes <- rownames(filtered)[grep("^ENS", rownames(filtered))]

y <- DGEList(counts=counts(Myset), group=as.factor(Design$TOI))
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
res <- residuals(fit, type="deviance")
Myset2 <- RUVr(Myset, genes, k=1, res)

ddsTxi <- DESeqDataSetFromTximport(txi,colData = pData(Myset2),design = ~ W_1 + TOI)
dds=DESeq(ddsTxi)
res=results(dds,contrast = c("TOI","Day_of_Discharge","Healthy"),independentFiltering = FALSE)                       
write.table(as.data.frame(res),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Result_RUV.txt",sep="\t", quote=FALSE,col.names = NA)

```

```{r,eval=FALSE}

## Healthy vs Day of Discharge

library(tximport)
library(tximportData)
library(DESeq2)
library(RUVSeq)
library(dplyr)

design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Design.txt",header = TRUE)
files <- file.path("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq",design$ID, "CodingCount.txt")
names(files) <- design$ID
txi <- tximport(files, type="salmon",txIn=FALSE,geneIdCol="Name")

count=as.data.frame(txi$counts)
count=round(count, 0)

Design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Design.txt",row.names = 1, check.names = FALSE)
filtered=count%>%filter(rowSums(across(where(is.numeric)))!=0)
Myset <- newSeqExpressionSet(as.matrix(filtered),phenoData = data.frame(Design))
Myset <- betweenLaneNormalization(Myset, which="upper")
design <- model.matrix(~TOI+Age+Gender+non_classical_monocyte+otherCells+memory_Bcell+intermediate_monocyte+MAIT_Tcell+plasmacytoid_DC, data=pData(Myset))
genes <- rownames(filtered)[grep("^ENS", rownames(filtered))]

y <- DGEList(counts=counts(Myset), group=as.factor(Design$TOI))
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
res <- residuals(fit, type="deviance")
Myset2 <- RUVr(Myset, genes, k=1, res)

ddsTxi <- DESeqDataSetFromTximport(txi,colData = pData(Myset2),design = ~ W_1 + TOI)
dds=DESeq(ddsTxi)
res=results(dds,contrast = c("TOI","One_month_after_discharge","Healthy"),independentFiltering = FALSE)                       
write.table(as.data.frame(res),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Result_RUV.txt",sep="\t", quote=FALSE,col.names = NA)

```

```{r,eval=FALSE}

## Day of Hospitalization vs Day of Discharge

library(tximport)
library(tximportData)
library(DESeq2)
library(RUVSeq)
library(dplyr)

design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_DOD/Design.txt",header = TRUE)
files <- file.path("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq",design$ID, "CodingCount.txt")
names(files) <- design$ID
txi <- tximport(files, type="salmon",txIn=FALSE,geneIdCol="Name")

count=as.data.frame(txi$counts)
count=round(count, 0)

Design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_DOD/Design.txt",row.names = 1, check.names = FALSE)
filtered=count%>%filter(rowSums(across(where(is.numeric)))!=0)
Myset <- newSeqExpressionSet(as.matrix(filtered),phenoData = data.frame(Design))
Myset <- betweenLaneNormalization(Myset, which="upper")
design <- model.matrix(~TOI+Age+Gender+non_classical_monocyte+classical_monocyte+intermediate_monocyte+neutrophil+Nkcell+plasmacytoid_DC, data=pData(Myset))
genes <- rownames(filtered)[grep("^ENS", rownames(filtered))]

y <- DGEList(counts=counts(Myset), group=as.factor(Design$TOI))
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
res <- residuals(fit, type="deviance")
Myset2 <- RUVr(Myset, genes, k=1, res)

ddsTxi <- DESeqDataSetFromTximport(txi,colData = pData(Myset2),design = ~ W_1 + Patient_Code+ TOI)
dds=DESeq(ddsTxi)
res=results(dds,contrast = c("TOI","Day_of_Discharge","Day_of_Hospitalization"),independentFiltering = FALSE)                       
write.table(as.data.frame(res),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_DOD/Result_RUV.txt",sep="\t", quote=FALSE,col.names = NA)

```


```{r,eval=FALSE}

## Day of Hospitalization vs One month after Discharge

library(tximport)
library(tximportData)
library(DESeq2)
library(RUVSeq)
library(dplyr)

design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Design.txt",header = TRUE)
files <- file.path("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq",design$ID, "CodingCount.txt")
names(files) <- design$ID
txi <- tximport(files, type="salmon",txIn=FALSE,geneIdCol="Name")

count=as.data.frame(txi$counts)
count=round(count, 0)

Design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Design.txt",row.names = 1, check.names = FALSE)
filtered=count%>%filter(rowSums(across(where(is.numeric)))!=0)
Myset <- newSeqExpressionSet(as.matrix(filtered),phenoData = data.frame(Design))
Myset <- betweenLaneNormalization(Myset, which="upper")
design <- model.matrix(~TOI+Age+Gender+plasmacytoid_DC+otherCells+intermediate_monocyte+neutrophil+classical_monocyte+naive_CD4_Tcell+Nkcell+memory_CD4_Tcell+non_classical_monocyte, data=pData(Myset))
genes <- rownames(filtered)[grep("^ENS", rownames(filtered))]

y <- DGEList(counts=counts(Myset), group=as.factor(Design$TOI))
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
res <- residuals(fit, type="deviance")
Myset2 <- RUVr(Myset, genes, k=1, res)

ddsTxi <- DESeqDataSetFromTximport(txi,colData = pData(Myset2),design = ~ W_1 + Patient_Code+ TOI)
dds=DESeq(ddsTxi)
res=results(dds,contrast = c("TOI","One_month_after_discharge","Day_of_Hospitalization"),independentFiltering = FALSE)                       
write.table(as.data.frame(res),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Result_RUV.txt",sep="\t", quote=FALSE,col.names = NA)

```


```{r,eval=FALSE}

## Day of Hospitalization vs One month after Discharge

library(tximport)
library(tximportData)
library(DESeq2)
library(RUVSeq)
library(dplyr)

design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOD_vs_OneMonth/Design.txt",header = TRUE)
files <- file.path("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq",design$ID, "CodingCount.txt")
names(files) <- design$ID
txi <- tximport(files, type="salmon",txIn=FALSE,geneIdCol="Name")

count=as.data.frame(txi$counts)
count=round(count, 0)

Design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOD_vs_OneMonth/Design.txt",row.names = 1, check.names = FALSE)
filtered=count%>%filter(rowSums(across(where(is.numeric)))!=0)
Myset <- newSeqExpressionSet(as.matrix(filtered),phenoData = data.frame(Design))
Myset <- betweenLaneNormalization(Myset, which="upper")
design <- model.matrix(~TOI+Age+Gender+plasmacytoid_DC+otherCells+MAIT_Tcell+myeloid_DC+non_classical_monocyte+memory_CD4_Tcell+intermediate_monocyte+naive_CD4_Tcell+memory_Bcell, data=pData(Myset))
genes <- rownames(filtered)[grep("^ENS", rownames(filtered))]

y <- DGEList(counts=counts(Myset), group=as.factor(Design$TOI))
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
res <- residuals(fit, type="deviance")
Myset2 <- RUVr(Myset, genes, k=1, res)

ddsTxi <- DESeqDataSetFromTximport(txi,colData = pData(Myset2),design = ~ W_1 + Patient_Code+ TOI)
dds=DESeq(ddsTxi)
res=results(dds,contrast = c("TOI","One_month_after_discharge","Day_of_Discharge"),independentFiltering = FALSE)                       
write.table(as.data.frame(res),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOD_vs_OneMonth/Result_RUV.txt",sep="\t", quote=FALSE,col.names = NA)

```

```{r,warning=FALSE,message=FALSE,eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOD_vs_OneMonth/Input_Piano.txt",header = TRUE,row.names = 1)
p <- data[5]
head(p)
lfc<-data[2]
head(lfc)
set.seed(123)

geneSets <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/KEGG_Selected/KEGG.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,
                 nPerm = 1000)
res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOD_vs_OneMonth/Piano_KEGG.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r,warning=FALSE,message=FALSE,eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_DOD/Input_Piano.txt",header = TRUE,row.names = 1)
p <- data[5]
head(p)
lfc<-data[2]
head(lfc)
set.seed(123)

geneSets <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/KEGG_Selected/KEGG.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,
                 nPerm = 1000)
res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_DOD/Piano_KEGG.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r,warning=FALSE,message=FALSE,eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Input_Piano.txt",header = TRUE,row.names = 1)
p <- data[5]
head(p)
lfc<-data[2]
head(lfc)
set.seed(123)

geneSets <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/KEGG_Selected/KEGG.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,
                 nPerm = 1000)
res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Piano_KEGG.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r,warning=FALSE,message=FALSE,eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Input_Piano.txt",header = TRUE,row.names = 1)
p <- data[5]
head(p)
lfc<-data[2]
head(lfc)
set.seed(123)

geneSets <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/KEGG_Selected/KEGG.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,
                 nPerm = 1000)
res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Piano_KEGG.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r,warning=FALSE,message=FALSE,eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Input_Piano.txt",header = TRUE,row.names = 1)
p <- data[5]
head(p)
lfc<-data[2]
head(lfc)
set.seed(123)

geneSets <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/KEGG_Selected/KEGG.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,
                 nPerm = 1000)
res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Piano_KEGG.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r,warning=FALSE,message=FALSE,eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Input_Piano.txt",header = TRUE,row.names = 1)
p <- data[5]
head(p)
lfc<-data[2]
head(lfc)
set.seed(123)

geneSets <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/KEGG_Selected/KEGG.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,
                 nPerm = 1000)
res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Piano_KEGG.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r, eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOD_vs_OneMonth/Reporter/Input_Piano.txt",header = TRUE,row.names = 2)

p <- data[6]
lfc<-data[3]
head(p)
head(lfc)

geneSets <- loadGSC(file = "/home/anoop/Desktop/COVID_Omics/Transcriptomics/DiffExp/Deseq/HC_Covid/HumanGEM_subsystem_GSC.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,geneSetStat="reporter",
                 nPerm = 1000)
 res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOD_vs_OneMonth/Reporter/ReporterSubSystem.txt",sep="\t",col.names = NA)

geneSets1 <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/GEM/FBA/Test/HumanGEM_metabolite_GSC.gmt")
gsares1 <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets1,geneSetStat="reporter",
                 nPerm = 1000)
res1=GSAsummaryTable(gsares1)
write.table(res1,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOD_vs_OneMonth/Reporter/ReporterMets.txt",sep="\t",col.names = NA)

```

```{r, eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_DOD/Reporter/Input_Piano.txt",header = TRUE,row.names = 2)

p <- data[6]
lfc<-data[3]
head(p)
head(lfc)

geneSets <- loadGSC(file = "/home/anoop/Desktop/COVID_Omics/Transcriptomics/DiffExp/Deseq/HC_Covid/HumanGEM_subsystem_GSC.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,geneSetStat="reporter",
                 nPerm = 1000)
 res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_DOD/Reporter/ReporterSubSystem.txt",sep="\t",col.names = NA)

geneSets1 <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/GEM/FBA/Test/HumanGEM_metabolite_GSC.gmt")
gsares1 <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets1,geneSetStat="reporter",
                 nPerm = 1000)
res1=GSAsummaryTable(gsares1)
write.table(res1,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_DOD/Reporter/ReporterMets.txt",sep="\t",col.names = NA)

```

```{r, eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Reporter/Input_Piano.txt",header = TRUE,row.names = 2)

p <- data[6]
lfc<-data[3]
head(p)
head(lfc)

geneSets <- loadGSC(file = "/home/anoop/Desktop/COVID_Omics/Transcriptomics/DiffExp/Deseq/HC_Covid/HumanGEM_subsystem_GSC.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,geneSetStat="reporter",
                 nPerm = 1000)
 res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Reporter/ReporterSubSystem.txt",sep="\t",col.names = NA)

geneSets1 <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/GEM/FBA/Test/HumanGEM_metabolite_GSC.gmt")
gsares1 <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets1,geneSetStat="reporter",
                 nPerm = 1000)
res1=GSAsummaryTable(gsares1)
write.table(res1,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Reporter/ReporterMets.txt",sep="\t",col.names = NA)

```

```{r, eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Reporter/Input_Piano.txt",header = TRUE,row.names = 2)

p <- data[6]
lfc<-data[3]
head(p)
head(lfc)

geneSets <- loadGSC(file = "/home/anoop/Desktop/COVID_Omics/Transcriptomics/DiffExp/Deseq/HC_Covid/HumanGEM_subsystem_GSC.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,geneSetStat="reporter",
                 nPerm = 1000)
 res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Reporter/ReporterSubSystem.txt",sep="\t",col.names = NA)

geneSets1 <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/GEM/FBA/Test/HumanGEM_metabolite_GSC.gmt")
gsares1 <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets1,geneSetStat="reporter",
                 nPerm = 1000)
res1=GSAsummaryTable(gsares1)
write.table(res1,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Reporter/ReporterMets.txt",sep="\t",col.names = NA)

```

```{r, eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Reporter/Input_Piano.txt",header = TRUE,row.names = 2)

p <- data[6]
lfc<-data[3]
head(p)
head(lfc)

geneSets <- loadGSC(file = "/home/anoop/Desktop/COVID_Omics/Transcriptomics/DiffExp/Deseq/HC_Covid/HumanGEM_subsystem_GSC.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,geneSetStat="reporter",
                 nPerm = 1000)
 res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Reporter/ReporterSubSystem.txt",sep="\t",col.names = NA)

geneSets1 <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/GEM/FBA/Test/HumanGEM_metabolite_GSC.gmt")
gsares1 <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets1,geneSetStat="reporter",
                 nPerm = 1000)
res1=GSAsummaryTable(gsares1)
write.table(res1,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Reporter/ReporterMets.txt",sep="\t",col.names = NA)

```


```{r, eval=FALSE}

library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Reporter/Input_Piano.txt",header = TRUE,row.names = 2)

p <- data[6]
lfc<-data[3]
head(p)
head(lfc)

geneSets <- loadGSC(file = "/home/anoop/Desktop/COVID_Omics/Transcriptomics/DiffExp/Deseq/HC_Covid/HumanGEM_subsystem_GSC.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,geneSetStat="reporter",
                 nPerm = 1000)
 res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Reporter/ReporterSubSystem.txt",sep="\t",col.names = NA)

geneSets1 <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/GEM/FBA/Test/HumanGEM_metabolite_GSC.gmt")
gsares1 <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets1,geneSetStat="reporter",
                 nPerm = 1000)
res1=GSAsummaryTable(gsares1)
write.table(res1,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Reporter/ReporterMets.txt",sep="\t",col.names = NA)

```

```{r,eval=FALSE}

## Impulse

library(tximport)
library(tximportData)
library(DESeq2)
library(RUVSeq)
library(dplyr)

design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/ImpulseDE/Design.txt",header = TRUE)
files <- file.path("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq",design$ID, "CodingCount.txt")
names(files) <- design$ID
txi <- tximport(files, type="salmon",txIn=FALSE,geneIdCol="Name")

count=as.data.frame(txi$counts)
count=round(count, 0)
write.table(count,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/ImpulseDE/Count.txt",sep="\t",col.names = NA)
```

```{r,eval=FALSE}
library(ImpulseDE2)
count=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/ImpulseDE/Count.txt",row.names = 1)
des=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/ImpulseDE/Design.txt",row.names = 1)
objectImpulseDE2 <- runImpulseDE2(
  matCountData    = as.matrix(count), 
  dfAnnotation    = des,
  boolCaseCtrl    = FALSE,
  vecConfounders  = NULL,
  boolIdentifyTransients = TRUE,
  scaNProc        = 14 )

write.table(objectImpulseDE2$dfImpulseDE2Results,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/ImpulseDE/Results.txt",sep="\t",col.names = NA)

```


```{r}
library(circlize)
library(ComplexHeatmap)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/Ip_Pwy.txt",header = TRUE,check.names = FALSE,row.names = 1)
head(data)

col_fun1 = colorRamp2(c(0,1.3,1.4,1.5,1.6,1.7,1.9,2.2), c("white" ,"#e5cccc","#cc9999","#bf7f7f","#a64c4c","#8c1919","#800000","#730000"))

sf=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/comparison.txt",header = TRUE,row.names = 1)
ha = HeatmapAnnotation(df = sf,show_annotation_name = FALSE,annotation_name_side = "right",simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.3, "cm"),
                                                      grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8), 
                                                      labels_gp = gpar(fontsize = 8)),
                       col = list(Comparison=c("HC_vs_T0"="#333333","HC_vs_T1"="#4c4c4c",
                                               "HC_vs_T2"="#737373","T0_vs_T1"="#999999","T0_vs_T2"="#bfbfbf","T1_vs_T2"="#d8d8d8"),
                                  Type=c("p adj (dist.dir.dn)"="#004C00","p adj (dist.dir.up)"="#e5c100")))

pwy=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/Pwy_type.txt",header = TRUE,row.names = 1)
ha2 = rowAnnotation(df = pwy,show_annotation_name = FALSE,simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.3, "cm"),
                                                      grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8), 
                                                      labels_gp = gpar(fontsize = 8)),
                       col = list(Category=c("Environmental Information Processing"="#a0a0a0","Metabolism"="#aa9980",
                                         "Organismal Systems"="#9ecfc2")))

H1=Heatmap(as.matrix(-log10(data)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,
           row_title_gp = gpar(fontsize=0),row_dend_width = unit(0.5, "cm"),left_annotation = ha2,row_split = pwy$Category,
           column_title_gp =gpar(fontsize = 0),top_annotation = ha,
           heatmap_legend_param =list(grid_width = unit(0.3, "cm"),grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8),
                                      labels_gp = gpar(fontsize = 8)),
           name = "-log10(Padj)",show_row_names = TRUE,row_names_gp=gpar(fontsize = 6),height  = unit(16.5, "cm"),
           width  = unit(4.5, "cm"),column_split = sf$Comparison)


pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/Pwy_type.pdf")
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "right",merge_legend = TRUE)
dev.off()
```

```{r}
library(circlize)
library(ComplexHeatmap)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/IP1.txt",header = TRUE,check.names = FALSE,row.names = 1)
head(data)

col_fun1 = colorRamp2(c(0,1.3,1.4,1.5,1.6,1.7,1.9,2.2), c("white" ,"#e5cccc","#cc9999","#bf7f7f","#a64c4c","#8c1919","#800000","#730000"))

sf=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/comparison.txt",header = TRUE,row.names = 1)
ha = HeatmapAnnotation(df = sf,show_annotation_name = FALSE,annotation_name_side = "right",simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.3, "cm"),
                                                      grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8), 
                                                      labels_gp = gpar(fontsize = 8)),
                       col = list(Comparison=c("HC_vs_T0"="#333333","HC_vs_T1"="#4c4c4c",
                                               "HC_vs_T2"="#737373","T0_vs_T1"="#999999","T0_vs_T2"="#bfbfbf","T1_vs_T2"="#d8d8d8"),
                                  Type=c("p adj (dist.dir.dn)"="#004C00","p adj (dist.dir.up)"="#e5c100")))

pwy=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/Pwy_type.txt",header = TRUE,row.names = 1)
ha2 = rowAnnotation(df = pwy,show_annotation_name = FALSE,simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.3, "cm"),
                                                      grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8), 
                                                      labels_gp = gpar(fontsize = 8)),
                       col = list(Category=c("Environmental Information Processing"="#a0a0a0","Metabolism"="#aa9980",
                                         "Organismal Systems"="#9ecfc2")))

H1=Heatmap(as.matrix(-log10(data)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,
           row_title_gp = gpar(fontsize=0),row_dend_width = unit(0.5, "cm"),
           column_title_gp =gpar(fontsize = 0),top_annotation = ha,
           heatmap_legend_param =list(grid_width = unit(0.3, "cm"),grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8),
                                      labels_gp = gpar(fontsize = 8)),
           name = "-log10(Padj)",show_row_names = TRUE,row_names_gp=gpar(fontsize =8),height  = unit(8, "cm"),
           width  = unit(4.5, "cm"),column_split = sf$Comparison)


pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/Pwy1.pdf")
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "right",merge_legend = TRUE)
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/IP2.txt",header = TRUE,check.names = FALSE,row.names = 1)
head(data)

col_fun1 = colorRamp2(c(0,1.3,1.4,1.5,1.6,1.7,1.9,2.2), c("white" ,"#e5cccc","#cc9999","#bf7f7f","#a64c4c","#8c1919","#800000","#730000"))

sf=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/comparison.txt",header = TRUE,row.names = 1)
ha = HeatmapAnnotation(df = sf,show_annotation_name = FALSE,annotation_name_side = "right",simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.3, "cm"),
                                                      grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8), 
                                                      labels_gp = gpar(fontsize = 8)),
                       col = list(Comparison=c("HC_vs_T0"="#333333","HC_vs_T1"="#4c4c4c",
                                               "HC_vs_T2"="#737373","T0_vs_T1"="#999999","T0_vs_T2"="#bfbfbf","T1_vs_T2"="#d8d8d8"),
                                  Type=c("p adj (dist.dir.dn)"="#004C00","p adj (dist.dir.up)"="#e5c100")))

pwy=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/Pwy_type.txt",header = TRUE,row.names = 1)
ha2 = rowAnnotation(df = pwy,show_annotation_name = FALSE,simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.3, "cm"),
                                                      grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8), 
                                                      labels_gp = gpar(fontsize = 8)),
                       col = list(Category=c("Environmental Information Processing"="#a0a0a0","Metabolism"="#aa9980",
                                         "Organismal Systems"="#9ecfc2")))

H1=Heatmap(as.matrix(-log10(data)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,
           row_title_gp = gpar(fontsize=0),row_dend_width = unit(0.5, "cm"),
           column_title_gp =gpar(fontsize = 0),top_annotation = ha,
           heatmap_legend_param =list(grid_width = unit(0.3, "cm"),grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8),
                                      labels_gp = gpar(fontsize = 8)),
           name = "-log10(Padj)",show_row_names = TRUE,row_names_gp=gpar(fontsize =8),height  = unit(10, "cm"),
           width  = unit(4.5, "cm"),column_split = sf$Comparison)


pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/Pwy2.pdf")
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "right",merge_legend = TRUE)
dev.off()
```

```{r}
library(circlize)
library(ComplexHeatmap)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/IP3.txt",header = TRUE,check.names = FALSE,row.names = 1)
head(data)

col_fun1 = colorRamp2(c(0,1.3,1.4,1.5,1.6,1.7,1.9,2.2), c("white" ,"#e5cccc","#cc9999","#bf7f7f","#a64c4c","#8c1919","#800000","#730000"))

sf=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/comparison.txt",header = TRUE,row.names = 1)
ha = HeatmapAnnotation(df = sf,show_annotation_name = FALSE,annotation_name_side = "right",simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.3, "cm"),
                                                      grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8), 
                                                      labels_gp = gpar(fontsize = 8)),
                       col = list(Comparison=c("HC_vs_T0"="#333333","HC_vs_T1"="#4c4c4c",
                                               "HC_vs_T2"="#737373","T0_vs_T1"="#999999","T0_vs_T2"="#bfbfbf","T1_vs_T2"="#d8d8d8"),
                                  Type=c("p adj (dist.dir.dn)"="#004C00","p adj (dist.dir.up)"="#e5c100")))

pwy=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/Pwy_type.txt",header = TRUE,row.names = 1)
ha2 = rowAnnotation(df = pwy,show_annotation_name = FALSE,simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.3, "cm"),
                                                      grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8), 
                                                      labels_gp = gpar(fontsize = 8)),
                       col = list(Category=c("Environmental Information Processing"="#a0a0a0","Metabolism"="#aa9980",
                                         "Organismal Systems"="#9ecfc2")))

H1=Heatmap(as.matrix(-log10(data)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,
           row_title_gp = gpar(fontsize=0),row_dend_width = unit(0.5, "cm"),
           column_title_gp =gpar(fontsize = 0),top_annotation = ha,
           heatmap_legend_param =list(grid_width = unit(0.3, "cm"),grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8),
                                      labels_gp = gpar(fontsize = 8)),
           name = "-log10(Padj)",show_row_names = TRUE,row_names_gp=gpar(fontsize =8),height  = unit(8, "cm"),
           width  = unit(4.5, "cm"),column_split = sf$Comparison)


pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Heatmap/Pathway/Pwy3.pdf")
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "right",merge_legend = TRUE)
dev.off()
```


```{r}
library(reshape)
R=read.delim("/home/anoop/Desktop/CCHF/GEM/FBA/Merged/Network/Network/NET/Filtered/results/Communities/Splitted/Degree/Figure/Mets.txt",header = TRUE)
head(R)
Cas=cast(R)
head(Cas)
write.table(t(Cas),file="/home/anoop/Desktop/CCHF/GEM/FBA/Merged/Network/Network/NET/Filtered/results/Communities/Splitted/Degree/Figure/MetsCasted.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r}
library(ComplexHeatmap)
library(circlize)

chrt <- read.delim("/home/anoop/Desktop/CCHF/GEM/FBA/Merged/Network/Network/NET/Filtered/results/Communities/Splitted/Degree/Figure/SamInfo.txt",row.names = 1,header = TRUE)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8), labels_gp = gpar(fontsize = 8)),
                              col = list(Group=c("HC"="#4682B4","T0"="#a70000",
                               "T1"="#e4b2b2","T2"="#738957")))

Mets=read.delim("/home/anoop/Desktop/CCHF/GEM/FBA/Merged/Network/Network/NET/Filtered/results/Communities/Splitted/Degree/Figure/MetsCasted.txt",row.names = 1)



col_gene = colorRamp2(c(0,60,70,100,150,200,500), c("#f2f2f2","#f3e4e4","#efdcdc","#af5252","#9c2727","#7c1f1f","#5d1717"))

#col_gene = colorRamp2(c(0,60,70,75,80,120), c("grey","#1c3448","#4682b4","#90b4d2","#7c1f1f","#5d1717"))


#col_gene = colorRamp2(c(0,60,70,80,90), c("grey","#1E90FF","#125699","#323232","#000000"))

H1=Heatmap(as.matrix((Mets)),col=col_gene,cluster_rows=TRUE,cluster_columns = FALSE,
           show_column_names = FALSE,row_title_gp =gpar(fontsize = 0),column_split = chrt$Group,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),column_names_gp =gpar(fontsize = 15),
           column_dend_height =unit(0.5, "cm"), 
           top_annotation = chrt_anno,clustering_distance_columns = "euclidean",
           row_gap = unit(2, "mm"),column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),at = c(0,100,200,500),
                                                                                          title_gp = gpar(fontsize = 8),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Degree",show_row_names = TRUE,
           row_names_gp=gpar(fontsize = 7),height  = unit(8, "cm"),width  = unit(12, "cm"))


pdf("/home/anoop/Desktop/CCHF/GEM/FBA/Merged/Network/Network/NET/Filtered/results/Communities/Splitted/Degree/Figure/MetsCasted.pdf")
draw(H1, merge_legend = TRUE,ht_gap = unit(0.3, "cm"),heatmap_legend_side = "right", annotation_legend_side = "bottom")
dev.off()

```


```{r}


dummy=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/dummy.txt",row.names = 1,header = TRUE)
  ggplot(data=dummy,aes(gdpPercap,lifeExp, color=as.factor(year))) +
  geom_point(aes(fill=as.factor(year)),size=3) +
  scale_x_log10()+
  geom_line(aes(group = paired),
            color="grey",
            arrow = arrow(type = "closed",
                          length=unit(0.075, "inches")))
```

```{r,eval=FALSE}
library(PCAtools)
count=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM.txt",row.names = 1,check.names = FALSE)
meta=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/Sample_Info.txt",row.names = 1)
p <- PCAtools::pca((count), metadata = meta, removeVar = 0.1,scale = TRUE)
write.table(p$rotated,file = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_PCA_Scaled.txt",sep = "\t",col.names = NA,quote = FALSE)
head(p$variance)

```

```{r}


data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_PCA.txt",row.names = 1,header = TRUE)

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_PCA.pdf")
ggplot(data, aes(PC1,PC2,color=Cohort))+geom_point(size=3,aes(fill=Cohort,color=Cohort,shape=SGS2))+
  scale_shape_manual(values=c("Mild"=22, "Moderate"=23,"Severe"=24))+theme_gray()+
    scale_color_manual(values=c("T0"="#740000","T1"="#ca6666","T2"="#315b7d"))+
    scale_fill_manual(values=c("T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"))+
  labs(x="PC1, 98.7% variation",y="PC2, 0.4% variation")+
  geom_line(data=data,aes(group = pair), color="#808080",arrow = arrow(type = "closed",length=unit(0.05, "inches")))+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = Cohort), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
  dev.off()
```

```{r}


data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_PCA.txt",row.names = 1,header = TRUE)

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_PCA2.pdf")
ggplot(data, aes(PC1,PC2,color=SGS2))+geom_point(size=2,aes(fill=SGS2,color=SGS2,shape=Cohort))+
  scale_shape_manual(values=c("T0"=22, "T1"=23,"T2"=21))+theme_bw()+
    scale_color_manual(values=c("Mild"="#b29a50","Moderate"="#b28500","Severe"="#634a00"))+
    scale_fill_manual(values=c("Mild"="#ffdc73","Moderate"="#ffbf00","Severe"="#a67c00"))+
  labs(x="PC1, 98.7% variation",y="PC2, 0.4% variation")+
  geom_line(data=data,aes(group = pair,color=SGS2),arrow = arrow(type = "closed",ends = "last",length=unit(0.05, "inches")))+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = SGS2), alpha = 0.25,linetype = "dotted",level = 0.8)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```


```{r}


data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_PCA.txt",row.names = 1,header = TRUE)

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_PCA2.pdf")
ggplot(data, aes(PC1,PC2,color=SGS2))+geom_point(size=3,aes(fill=SGS2,color=SGS2,shape=Cohort))+
  scale_shape_manual(values=c("T0"=22, "T1"=23,"T2"=21))+theme_gray()+
    scale_color_manual(values=c("Mild"="#740000","Moderate"="#ca6666"))+
    scale_fill_manual(values=c("Mild"="#a70000","Moderate"="#e4b2b2"))+
  labs(x="PC1, 98.7% variation",y="PC2, 0.4% variation")+
  geom_line(data=data,aes(group = pair), color="black",arrow = arrow(type = "closed",ends = "last",length=unit(0.05, "inches")))+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = SGS2), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```


```{r}


data=read.delim("/home/anoop/Desktop/CCHF/SwimmerPlot/data1.txt")


try=read.delim("/home/anoop/Desktop/CCHF/SwimmerPlot/try.txt")

line=read.delim("/home/anoop/Desktop/CCHF/SwimmerPlot/line.txt")
st=read.delim("/home/anoop/Desktop/CCHF/SwimmerPlot/st.txt")
pdf("/home/anoop/Desktop/CCHF/SwimmerPlot/test.pdf")
swimmer_plot(df=data,id='id',end='Day',name_fill='as.factor(SGS)',
                         col="white",width=1,id_order = c("20","9","3","7","6","21","8","17","22","5","15","19","1","16","13","12","23","4","31","14","18","2","30","26","28","32","24","25","29","27"))+
  scale_fill_manual(values=c("0"="#cccccc","1"="#fff9f5","2"="#fff3ec","3"="#ffede2","4"="#ffe7d9","5"="#ffe1cf","6"="#ffdbc6",
"7"="#ffd5bc","8"="#ffcfb3","9"="#ffc3a0","10"="#cc9c80","11"="#997560","12"="#7f6150","13"="#664e40"))+
  swimmer_points(df_points=try,id='id',time='time',name_shape =
 'sampling',size=1.75,fill='white',col='#056273')+scale_shape_manual(values = c("T0"=16,"T1"=17,"T2"=15,"Survived"=8))+ 
  swimmer_lines(df_lines = line,id='id',start = 'start',end='end',
                name_col='group',size=5)+scale_color_manual(values=c("Mild"="#ffdc73","Moderate"="#ffbf00","Severe"="#a67c00"))+
  theme(plot.margin = margin(4.5,1,4.5,1, "cm"),axis.text = element_text(size=8,colour = "black"))
dev.off()

?swimmer_arrows
```

```{r}
head(ClinicalTrial.Response)
library(swimplot)
data=read.delim("/home/anoop/Desktop/CCHF/SwimmerPlot/data1.txt")

try=read.delim("/home/anoop/Desktop/CCHF/SwimmerPlot/try.txt")

oth=read.delim("/home/anoop/Desktop/CCHF/SwimmerPlot/other.txt")
line=read.delim("/home/anoop/Desktop/CCHF/SwimmerPlot/line.txt")

st=read.delim("/home/anoop/Desktop/CCHF/SwimmerPlot/st.txt")

pdf("/home/anoop/Desktop/CCHF/SwimmerPlot/test.pdf")
swimmer_plot(df=data,id='id',end='Day',name_fill='as.factor(SGS)',
                         col="white",width=1,id_order = c("20","9","3","7","6","21","8","17","22","5","15","19","1","16","13","12","23","4",
                                                          "31","14","18","2","30","26","28","32","24","25","29","27"))+
  scale_fill_manual(values=c("0"="#cccccc","1"="#fff7f8","2"="#fff0f2","3"="#ffe9ec",
                             "4"="#ffe1e6","5"="#ffdae0","6"="#ffd3d9",
"7"="#ffcbd3","8"="#ffbdc7","9"="#e5a3ad","10"="#b27f87","11"="#996d73",
"12"="#7f5b60","13"="#66484d"))+
  swimmer_lines(df_lines = line,id='id',start = 'start',end='end',
                name_col='group',size=3)+
  scale_color_manual(values=c("Mild"="#ffdc73","Moderate"="#ffbf00","Severe"="#a67c00","onset"="red"))+
  scale_y_continuous(limits = c(-7, 40), breaks = seq(-7, 40, by = 2))+
  swimmer_points(df_points=try,id='id',time='time',name_shape =
 'sampling',size=1.75,fill='white',col='#056273')+scale_shape_manual(values = c("T0"=16,"T1"=17,"T2"=15,"Survived"=8))+ 
  theme(plot.margin = margin(4.5,1,4.5,1, "cm"),axis.text = element_text(size=5,colour = "black"))
dev.off()
```

```{r, eval=FALSE}
set.seed(123)
library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Reporter/Input_Piano.txt",header = TRUE,row.names = 2)

p <- data[6]
lfc<-data[3]

geneSets1 <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/GEM/FBA/Test/HumanGEM_metabolite_GSC.gmt")
gsares1 <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets1,geneSetStat="reporter",
                 nPerm = 1000)
res1=GSAsummaryTable(gsares1)

write.table(res1,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Reporter/ReporterMets.txt",sep="\t",col.names = NA)

NET=networkPlot2(gsares1,class = "distinct",direction = "both",adjusted = TRUE,significance = 1,overlap = 5)

write.table(as.data.frame(NET$x$nodes),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Reporter/Nodes.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(as.data.frame(NET$x$edges),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Reporter/Edges.txt",sep="\t",col.names = NA,quote = FALSE)


#####

geneSets <- loadGSC(file = "/home/anoop/Desktop/COVID_Omics/Transcriptomics/DiffExp/Deseq/HC_Covid/HumanGEM_subsystem_GSC.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,geneSetStat="reporter",
                 nPerm = 1000)
 res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Reporter/ReporterSubSystem.txt",sep="\t",col.names = NA)


NET1=networkPlot2(gsares,class = "distinct",direction = "both",adjusted = TRUE,significance = 0.2,overlap = 5)

write.table(as.data.frame(NET1$x$nodes),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Reporter/NodesSub.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(as.data.frame(NET1$x$edges),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Reporter/EdgesSub.txt",sep="\t",col.names = NA,quote = FALSE)

```


```{r, eval=FALSE}
set.seed(123)
library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Reporter/Input_Piano.txt",header = TRUE,row.names = 2)

p <- data[6]
lfc<-data[3]

geneSets1 <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/GEM/FBA/Test/HumanGEM_metabolite_GSC.gmt")
gsares1 <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets1,geneSetStat="reporter",
                 nPerm = 1000)
res1=GSAsummaryTable(gsares1)

write.table(res1,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Reporter/ReporterMets.txt",sep="\t",col.names = NA)

NET=networkPlot2(gsares1,class = "distinct",direction = "both",adjusted = TRUE,significance = 1,overlap = 5)

write.table(as.data.frame(NET$x$nodes),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Reporter/Nodes.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(as.data.frame(NET$x$edges),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Reporter/Edges.txt",sep="\t",col.names = NA,quote = FALSE)


#####

geneSets <- loadGSC(file = "/home/anoop/Desktop/COVID_Omics/Transcriptomics/DiffExp/Deseq/HC_Covid/HumanGEM_subsystem_GSC.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,geneSetStat="reporter",
                 nPerm = 1000)
 res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Reporter/ReporterSubSystem.txt",sep="\t",col.names = NA)


NET1=networkPlot2(gsares,class = "distinct",direction = "both",adjusted = TRUE,significance = 0.2,overlap = 5)

write.table(as.data.frame(NET1$x$nodes),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Reporter/NodesSub.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(as.data.frame(NET1$x$edges),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Reporter/EdgesSub.txt",sep="\t",col.names = NA,quote = FALSE)

```


```{r, eval=FALSE}
set.seed(123)
library(piano)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Reporter/Input_Piano.txt",header = TRUE,row.names = 2)

p <- data[6]
lfc<-data[3]

geneSets1 <- loadGSC(file = "/home/anoop/Desktop/Transcriptome/New_RNAseq/Project2_ART/With New HC/GEM/FBA/Test/HumanGEM_metabolite_GSC.gmt")
gsares1 <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets1,geneSetStat="reporter",
                 nPerm = 1000)
res1=GSAsummaryTable(gsares1)

write.table(res1,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Reporter/ReporterMets.txt",sep="\t",col.names = NA)

NET=networkPlot2(gsares1,class = "distinct",direction = "both",adjusted = TRUE,significance = 1,overlap = 5)

write.table(as.data.frame(NET$x$nodes),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Reporter/Nodes.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(as.data.frame(NET$x$edges),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Reporter/Edges.txt",sep="\t",col.names = NA,quote = FALSE)


#####

geneSets <- loadGSC(file = "/home/anoop/Desktop/COVID_Omics/Transcriptomics/DiffExp/Deseq/HC_Covid/HumanGEM_subsystem_GSC.gmt")
gsares <- runGSA(geneLevelStats=p,directions=lfc,adjMethod = "BH",
                 gsc = geneSets,geneSetStat="reporter",
                 nPerm = 1000)
 res=GSAsummaryTable(gsares)
write.table(res,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Reporter/ReporterSubSystem.txt",sep="\t",col.names = NA)


NET1=networkPlot2(gsares,class = "distinct",direction = "both",adjusted = TRUE,significance = 0.2,overlap = 5)

write.table(as.data.frame(NET1$x$nodes),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Reporter/NodesSub.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(as.data.frame(NET1$x$edges),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Reporter/EdgesSub.txt",sep="\t",col.names = NA,quote = FALSE)

```


```{r}
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/GSEA/Result/Figure.txt",row.names = 1)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/GSEA/Result/Figure.pdf")
ggplot(dat, aes(x=order,y=NES))+geom_point(data=subset(dat,FDR>0.2),aes(size=lfdr),fill="grey",color="#666666",shape=21)+
  geom_point(data=subset(dat,FDR<=0.2 & NES>0),aes(size=lfdr),fill="#b20000",color="#6a0000",shape=21)+
  geom_point(data=subset(dat,FDR<=0.2 & NES<0),aes(size=lfdr),fill="#004c00",color="#003c00",shape=21)+
  geom_label_repel(aes(label = Label2),segment.color = '#cccccc',fill=NA,color="black",size=2.5,
                   label.size = NA,segment.alpha=0.75,box.padding=0,nudge_x = 0.75,nudge_y = -0.75)+
  geom_label_repel(aes(label = Label3),segment.color = '#cccccc',fill=NA,color="black",size=2.5,
                   label.size = NA,segment.alpha=0.75,box.padding=0,nudge_x = 0.75,nudge_y = -1)+
  theme(axis.title.x = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank(),legend.position = "right",plot.margin = margin(5.5,2.5,5.5,2.5, "cm"),plot.title = element_text(hjust = 0.5,size = 12),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  guides(size = guide_legend(ncol = 1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol = 1),fill = guide_legend(ncol = 1))
dev.off()
```


```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/WithGeneSymbol_Result_RUV2.txt",row.names = 1)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/WithGeneSymbol_Result_RUV2.pdf")
library(ggrepel)
ggplot(data, aes(x=log2(baseMean), y=log2FoldChange)) + 
  geom_point(data=subset(data, padj>=0.05),aes(x=log2(baseMean), y=log2FoldChange,size=-log10(padj)),color="#bfbfbf")+
  geom_point(data=subset(data, padj<0.05 & log2FoldChange < 0),aes(x=log2(baseMean),y=log2FoldChange,
                                                                   color=abs(log2FoldChange),size=-log10(padj)))+
  scale_color_gradient(low = "#e3fc03", high = "#004c00")+
  geom_point(data=subset(data, padj<0.05 & log2FoldChange > 0),shape=21,
             aes(x=log2(baseMean), y=log2FoldChange,fill=abs(log2FoldChange),
                 size=-log10(padj)),color="transparent")+
  scale_fill_gradient(low = "yellow", high = "#b20000")+
  scale_size(range = c(0.1, 4))+
  scale_y_continuous(limits = c(-6, 9.5), breaks = seq(-9.5, 9.5, by = 3))+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2.5,
                   label.size = NA,segment.alpha=0.75,box.padding=0,nudge_x = 0.55,nudge_y = -0.5,max.overlaps = 200)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),axis.title.y=element_text(size=8),legend.position = "none",
        axis.title.x=element_text(size=8),axis.text=element_text(size=8,color="black"),plot.margin = margin(5,3.5,5,3, "cm"))+
   labs(x="Log2(Base Mean)",y="Log2 Fold Change")
dev.off()
```


```{r}
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/GSEA/Figure.txt",row.names = 1)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/GSEA/Figure.pdf")
ggplot(dat, aes(x=order,y=NES))+geom_point(data=subset(dat,FDR>0.2),aes(size=lfdr),fill="grey",color="#666666",shape=21)+
  geom_point(data=subset(dat,FDR<=0.2 & NES>0),aes(size=lfdr),fill="#b20000",color="#6a0000",shape=21)+
  geom_point(data=subset(dat,FDR<=0.2 & NES<0),aes(size=lfdr),fill="#004c00",color="#003c00",shape=21)+
  geom_label_repel(aes(label = Label2),segment.color = '#cccccc',fill=NA,color="black",size=2.5,
                   label.size = NA,segment.alpha=0.75,box.padding=0,nudge_x = 0.75,nudge_y = 0.75)+
  theme(axis.title.x = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank(),legend.position = "right",plot.margin = margin(5.5,2.5,5.5,2.5, "cm"),plot.title = element_text(hjust = 0.5,size = 12),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  guides(size = guide_legend(ncol = 1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol = 1),fill = guide_legend(ncol = 1))
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/WithGeneSymbol_Result_RUV2.txt",row.names = 1)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/WithGeneSymbol_Result_RUV2.pdff")
library(ggrepel)
ggplot(data, aes(x=log2(baseMean), y=log2FoldChange)) + 
  geom_point(data=subset(data, padj>=0.05),aes(x=log2(baseMean), y=log2FoldChange,size=-log10(padj)),color="#bfbfbf")+
  geom_point(data=subset(data, padj<0.05 & log2FoldChange < 0),aes(x=log2(baseMean),y=log2FoldChange,
                                                                   color=abs(log2FoldChange),size=-log10(padj)))+
  scale_color_gradient(low = "#e3fc03", high = "#004c00")+
  geom_point(data=subset(data, padj<0.05 & log2FoldChange > 0),shape=21,
             aes(x=log2(baseMean), y=log2FoldChange,fill=abs(log2FoldChange),
                 size=-log10(padj)),color="transparent")+
  scale_fill_gradient(low = "yellow", high = "#b20000")+
  scale_size(range = c(0.1, 4))+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2.5,
                   label.size = NA,segment.alpha=0.75,box.padding=0,nudge_x = 0.55,nudge_y = -0.5,max.overlaps = 700)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),axis.title.y=element_text(size=8),legend.position = "none",
        axis.title.x=element_text(size=8),axis.text=element_text(size=8,color="black"),plot.margin = margin(5,3.5,5,3, "cm"))+
   labs(x="Log2(Base Mean)",y="Log2 Fold Change")
dev.off()
```


```{r}
dat=read.delim("//home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/GSEA/Figure.txt",row.names = 1)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/GSEA/Figure.pdf")
ggplot(dat, aes(x=order,y=NES))+geom_point(data=subset(dat,FDR>0.2),aes(size=lfdr),fill="grey",color="#666666",shape=21)+
  geom_point(data=subset(dat,FDR<=0.2 & NES>0),aes(size=lfdr),fill="#b20000",color="#6a0000",shape=21)+
  geom_point(data=subset(dat,FDR<=0.2 & NES<0),aes(size=lfdr),fill="#004c00",color="#003c00",shape=21)+
  geom_label_repel(aes(label = Label2),segment.color = '#cccccc',fill=NA,color="black",size=2.5,
                   label.size = NA,segment.alpha=0.75,box.padding=0,nudge_x = 0.75,nudge_y = 0.75)+
  theme(axis.title.x = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank(),legend.position = "right",plot.margin = margin(5.5,2.5,5.5,2.5, "cm"),plot.title = element_text(hjust = 0.5,size = 12),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  guides(size = guide_legend(ncol = 1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol = 1),fill = guide_legend(ncol = 1))
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/WithGeneSymbol_Result_RUV2.txt",row.names = 1)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/WithGeneSymbol_Result_RUV2.pdf")
library(ggrepel)
ggplot(data, aes(x=log2(baseMean), y=log2FoldChange)) + 
  geom_point(data=subset(data, padj>=0.05),aes(x=log2(baseMean), y=log2FoldChange,size=-log10(padj)),color="#bfbfbf")+
  geom_point(data=subset(data, padj<0.05 & log2FoldChange < 0),aes(x=log2(baseMean),y=log2FoldChange,
                                                                   color=abs(log2FoldChange),size=-log10(padj)))+
  scale_color_gradient(low = "#e3fc03", high = "#004c00")+
  geom_point(data=subset(data, padj<0.05 & log2FoldChange > 0),shape=21,
             aes(x=log2(baseMean), y=log2FoldChange,fill=abs(log2FoldChange),
                 size=-log10(padj)),color="transparent")+
  scale_fill_gradient(low = "yellow", high = "#b20000")+
  scale_size(range = c(0.1, 4))+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2.5,
                   label.size = NA,segment.alpha=0.75,box.padding=0,nudge_x = 0.55,nudge_y = -0.5,max.overlaps = 700)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),axis.title.y=element_text(size=8),legend.position = "none",
        axis.title.x=element_text(size=8),axis.text=element_text(size=8,color="black"),plot.margin = margin(5,3.5,5,3, "cm"))+
   labs(x="Log2(Base Mean)",y="Log2 Fold Change")
dev.off()
```

```{r}
library(reshape2)
df=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Time/Decreasing.txt")
M=melt(df)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Time/Decreasing.pdf")
ggplot(M,aes(x=factor(variable,levels = unique(variable)),y=log(value),fill=variable,color=variable))+geom_boxplot(outlier.shape = NA)+
  facet_wrap(~ variable, ncol = 4,scales ="free")+
scale_color_manual(values=c("T2"="#315b7d","T0"="#740000",
                                "T1"="#ca6666"))+
    scale_fill_manual(values=c("T2"="#4682B4","T0"="#a70000",
                               "T1"="#e4b2b2"))+
  theme_bw()+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank(),legend.position ="bottom",panel.border = element_blank(),
        axis.ticks.x = element_blank(),axis.text.x = element_blank(),legend.title = element_blank(),legend.text = element_text(size = 6),
        plot.margin = margin(6,6,6,6, "cm"),strip.text.x = element_text(size = 6),
        axis.text.y = element_text(color="black",size=6),plot.title = element_text(hjust = 0.5,size = 6))+
  guides(color = guide_legend(nrow = 1),fill = guide_legend(nrow = 1))
dev.off()
```

```{r}
library(reshape2)
df=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Time/DecreasingIncreasing.txt")
M=melt(df)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Time/DecreasingIncreasing.pdf")
ggplot(M,aes(x=factor(variable,levels = unique(variable)),y=log(value),fill=variable,color=variable))+geom_boxplot(outlier.shape = NA)+
  facet_wrap(~ variable, ncol = 4,scales ="free")+
scale_color_manual(values=c("T2"="#315b7d","T0"="#740000",
                                "T1"="#ca6666"))+
    scale_fill_manual(values=c("T2"="#4682B4","T0"="#a70000",
                               "T1"="#e4b2b2"))+
  theme_bw()+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank(),legend.position ="bottom",panel.border = element_blank(),
        axis.ticks.x = element_blank(),axis.text.x = element_blank(),legend.title = element_blank(),legend.text = element_text(size = 6),
        plot.margin = margin(6,6,6,6, "cm"),strip.text.x = element_text(size = 6),
        axis.text.y = element_text(color="black",size=6),plot.title = element_text(hjust = 0.5,size = 6))+
  guides(color = guide_legend(nrow = 1),fill = guide_legend(nrow = 1))
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Time/kegg/Decreasing/Ip.txt",sep = "\t")
data$Term <- factor(data$Term, levels = data$Term)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Time/kegg/Decreasing/Ip.pdf")
ggplot(data, aes(y=Term)) + theme_bw()+
  geom_point(data=data,aes(x=1,y=Term,size=Overlap,color=lpval))+scale_x_discrete(limits=c("1"))+
  scale_color_gradient(low="#669cc0",high="#00365a")+
  scale_y_discrete(position = "left")+
  theme(axis.title = element_blank(),axis.text.x = element_blank(),axis.text.y = element_text(size=8,color="black"),
        axis.ticks = element_blank(),legend.position = "right",legend.box="vertical",
        legend.key.size = unit(0.7,"line"),legend.text = element_text(size=10,colour = "black"),plot.margin = margin(3.5,4,4,4, "cm"),
        legend.title = element_text(size=10))+
  guides(size=guide_legend(override.aes=list(colour="grey"),title = "Overlap",ncol = 1),color=guide_legend(title = "-log10(padj)",ncol = 1,override.aes = list(size = 5)))
dev.off()
```


```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Time/kegg/IncreasingDecreasing/Ip.txt",sep = "\t")
data$Term <- factor(data$Term, levels = data$Term)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Time/kegg/IncreasingDecreasing/Ip.pdf")
ggplot(data, aes(y=Term)) + theme_bw()+
  geom_point(data=data,aes(x=1,y=Term,size=Overlap,color=lpval))+scale_x_discrete(limits=c("1"))+
  scale_color_gradient(low="#669cc0",high="#00365a")+
  scale_y_discrete(position = "left")+
  theme(axis.title = element_blank(),axis.text.x = element_blank(),axis.text.y = element_text(size=8,color="black"),
        axis.ticks = element_blank(),legend.position = "right",legend.box="vertical",
        legend.key.size = unit(0.7,"line"),legend.text = element_text(size=10,colour = "black"),plot.margin = margin(3.5,3,4,4, "cm"),
        legend.title = element_text(size=10))+
  guides(size=guide_legend(override.aes=list(colour="grey"),title = "Overlap",ncol = 1),color=guide_legend(title = "-log10(padj)",ncol = 1,override.aes = list(size = 5)))
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Group=c("HC"="#4682B4","T0"="#a70000","T1"="#e4b2b2","T2"="#738957")))


FBA=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/TransT0.txt",row.names = 1)


col_fun1 = colorRamp2(c(1000,750,500,250, 0,-250,-500,750,-1000), c("#333300","#666600","#e5e500","#ffffcc","white","#d8b2d8","#a64ca6","#730073","#400040"))


H1=Heatmap(as.matrix((FBA)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),column_split = chrt$Group,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           top_annotation = chrt_anno,clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1000,750,500,250,0,-250,-500,-750,-1000),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Flux",show_row_names = TRUE,row_names_gp=gpar(fontsize =8),
           height  = unit(7, "cm"),width  = unit(2, "cm"))

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/TransT0.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="left", annotation_legend_side="left")
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Group=c("HC"="#4682B4","T0"="#a70000","T1"="#e4b2b2","T2"="#738957")))


FBA=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/OtherT0.txt",row.names = 1)


col_fun1 = colorRamp2(c(1000,750,500,250, 0,-250,-500,750,-1000), c("#333300","#666600","#e5e500","#ffffcc","white","#d8b2d8","#a64ca6","#730073","#400040"))


H1=Heatmap(as.matrix((FBA)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),column_split = chrt$Group,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           top_annotation = chrt_anno,clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1000,750,500,250,0,-250,-500,-750,-1000),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Flux",show_row_names = TRUE,row_names_gp=gpar(fontsize = 6),
           height  = unit(12, "cm"),width  = unit(2, "cm"))

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/OtherT0.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Group=c("HC"="#4682B4","T0"="#a70000","T1"="#e4b2b2","T2"="#738957")))


FBA=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/TransT2.txt",row.names = 1)


col_fun1 = colorRamp2(c(1000,750,500,250, 0,-250,-500,750,-1000), c("#333300","#666600","#e5e500","#ffffcc","white","#d8b2d8","#a64ca6","#730073","#400040"))


H1=Heatmap(as.matrix((FBA)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),column_split = chrt$Group,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           top_annotation = chrt_anno,clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1000,750,500,250,0,-250,-500,-750,-1000),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Flux",show_row_names = TRUE,row_names_gp=gpar(fontsize =8),
           height  = unit(7, "cm"),width  = unit(2, "cm"))

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/TransT2.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="left", annotation_legend_side="left")
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Group=c("HC"="#4682B4","T0"="#a70000","T1"="#e4b2b2","T2"="#738957")))


FBA=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/OtherT2.txt",row.names = 1)


col_fun1 = colorRamp2(c(1000,750,500,250, 0,-250,-500,750,-1000), c("#333300","#666600","#e5e500","#ffffcc","white","#d8b2d8","#a64ca6","#730073","#400040"))


H1=Heatmap(as.matrix((FBA)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),column_split = chrt$Group,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           top_annotation = chrt_anno,clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1000,750,500,250,0,-250,-500,-750,-1000),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Flux",show_row_names = TRUE,row_names_gp=gpar(fontsize = 6),
           height  = unit(5, "cm"),width  = unit(2, "cm"))

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/OtherT2.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Group=c("HC"="#4682B4","T0"="#a70000","T1"="#e4b2b2","T2"="#738957")))


FBA=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Neg/TransT0.txt",row.names = 1)


col_fun1 = colorRamp2(c(1000,750,500,250, 0,-250,-500,750,-1000), c("#333300","#666600","#e5e500","#ffffcc","white","#d8b2d8","#a64ca6","#730073","#400040"))


H1=Heatmap(as.matrix((FBA)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),column_split = chrt$Group,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           top_annotation = chrt_anno,clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1000,750,500,250,0,-250,-500,-750,-1000),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Flux",show_row_names = TRUE,row_names_gp=gpar(fontsize =8),
           height  = unit(4, "cm"),width  = unit(2, "cm"))

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Neg/TransT0.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="left", annotation_legend_side="left")
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Group=c("HC"="#4682B4","T0"="#a70000","T1"="#e4b2b2","T2"="#738957")))


FBA=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Neg/TransT2.txt",row.names = 1)


col_fun1 = colorRamp2(c(1000,750,500,250, 0,-250,-500,750,-1000), c("#333300","#666600","#e5e500","#ffffcc","white","#d8b2d8","#a64ca6","#730073","#400040"))


H1=Heatmap(as.matrix((FBA)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),column_split = chrt$Group,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           top_annotation = chrt_anno,clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1000,750,500,250,0,-250,-500,-750,-1000),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Flux",show_row_names = TRUE,row_names_gp=gpar(fontsize = 6),
           height  = unit(4, "cm"),width  = unit(2, "cm"))

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Neg/TransT2.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Group=c("HC"="#4682B4","T0"="#a70000","T1"="#e4b2b2","T2"="#738957")))


FBA=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Neg/OtherT0.txt",row.names = 1)


col_fun1 = colorRamp2(c(1000,750,500,250, 0,-250,-500,750,-1000), c("#333300","#666600","#e5e500","#ffffcc","white","#d8b2d8","#a64ca6","#730073","#400040"))


H1=Heatmap(as.matrix((FBA)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),column_split = chrt$Group,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           top_annotation = chrt_anno,clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1000,750,500,250,0,-250,-500,-750,-1000),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Flux",show_row_names = TRUE,row_names_gp=gpar(fontsize = 6),
           height  = unit(4, "cm"),width  = unit(2, "cm"))

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Neg/OtherT0.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Pos/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Group=c("HC"="#4682B4","T0"="#a70000","T1"="#e4b2b2","T2"="#738957")))


FBA=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Neg/OtherT2.txt",row.names = 1)


col_fun1 = colorRamp2(c(1000,750,500,250, 0,-250,-500,750,-1000), c("#333300","#666600","#e5e500","#ffffcc","white","#d8b2d8","#a64ca6","#730073","#400040"))


H1=Heatmap(as.matrix((FBA)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),column_split = chrt$Group,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           top_annotation = chrt_anno,clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1000,750,500,250,0,-250,-500,-750,-1000),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Flux",show_row_names = TRUE,row_names_gp=gpar(fontsize = 6),
           height  = unit(4, "cm"),width  = unit(2, "cm"))

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Venn/Neg/OtherT2.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/WithGeneSymbol_Result_RUV2.txt",row.names = 1)

library(ggrepel)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOH/Volcano_HC_T0.pdf")
ggplot(data, aes(x=log2FoldChange, y=-log10(padj))) +
  geom_point(data=subset(data, padj <.05 & log2FoldChange <= -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#166b99",fill="#26b3ff",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange >= 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,fill="#ffad73",color="#996745",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange > 0 & log2FoldChange < 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#ffcdab",fill="#ffe6d5",size=1)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange < 0 & log2FoldChange > -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#7cd1ff",fill="#bde8ff",size=1)+
  geom_point(data=subset(data, padj >=.05),aes(x=log2FoldChange,y=-log10(padj)),pch=21,color="#8c8c8c",fill="#cccccc",size=0.5)+
  geom_vline(xintercept=1, linetype="dashed",size=0.35)+
  geom_vline(xintercept=-1, linetype="dashed",size=0.35)+scale_x_continuous(limits = c(-8, 10), breaks = seq(-8, 10, by = 2))+
  geom_hline(yintercept=1.3010299957, linetype="dashed",size=0.35)+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2,label.size = NA,segment.alpha=0.5,
                   box.padding=0.8,nudge_x = 0.1,nudge_y = 0.1,max.overlaps = 2000)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),
        axis.title=element_text(size=10),axis.text.y=element_text(size=10),axis.text.x=element_text(size=10),
        plot.margin = margin(4.5,4,4.5,4, "cm"))+
  labs(x="Log2 Fold Change",y="-log10 (adj.p)")
dev.off()
```
```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/WithGeneSymbol_Result_RUV2.txt",row.names = 1)

library(ggrepel)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_DOD/Volcano_HC_T1.pdf")
ggplot(data, aes(x=log2FoldChange, y=-log10(padj))) +
  geom_point(data=subset(data, padj <.05 & log2FoldChange <= -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#166b99",fill="#26b3ff",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange >= 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,fill="#ffad73",color="#996745",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange > 0 & log2FoldChange < 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#ffcdab",fill="#ffe6d5",size=1)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange < 0 & log2FoldChange > -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#7cd1ff",fill="#bde8ff",size=1)+
  geom_point(data=subset(data, padj >=.05),aes(x=log2FoldChange,y=-log10(padj)),pch=21,color="#8c8c8c",fill="#cccccc",size=0.5)+
  geom_vline(xintercept=1, linetype="dashed",size=0.35)+
  geom_vline(xintercept=-1, linetype="dashed",size=0.35)+scale_x_continuous(limits = c(-30, 30), breaks = seq(-30, 30, by = 10))+
  geom_hline(yintercept=1.3010299957, linetype="dashed",size=0.35)+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2,label.size = NA,segment.alpha=0.5,
                   box.padding=0.8,nudge_x = 0.1,nudge_y = 0.1,max.overlaps = 2000)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),
        axis.title=element_text(size=10),axis.text.y=element_text(size=10),axis.text.x=element_text(size=10),
        plot.margin = margin(4.5,4,4.5,4, "cm"))+
  labs(x="Log2 Fold Change",y="-log10 (adj.p)")
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/WithGeneSymbol_Result_RUV2.txt",row.names = 1)

library(ggrepel)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/Volcano_HC_T2.pdf")
ggplot(data, aes(x=log2FoldChange, y=-log10(padj))) +
  geom_point(data=subset(data, padj <.05 & log2FoldChange <= -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#166b99",fill="#26b3ff",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange >= 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,fill="#ffad73",color="#996745",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange > 0 & log2FoldChange < 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#ffcdab",fill="#ffe6d5",size=1)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange < 0 & log2FoldChange > -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#7cd1ff",fill="#bde8ff",size=1)+
  geom_point(data=subset(data, padj >=.05),aes(x=log2FoldChange,y=-log10(padj)),pch=21,color="#8c8c8c",fill="#cccccc",size=0.5)+
  geom_vline(xintercept=1, linetype="dashed",size=0.35)+
  geom_vline(xintercept=-1, linetype="dashed",size=0.35)+scale_x_continuous(limits = c(-6, 6), breaks = seq(-6, 6, by = 2))+
  geom_hline(yintercept=1.3010299957, linetype="dashed",size=0.35)+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2,label.size = NA,segment.alpha=0.5,
                   box.padding=0.8,nudge_x = 0.1,nudge_y = 0.1,max.overlaps = 2000)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),
        axis.title=element_text(size=10),axis.text.y=element_text(size=10),axis.text.x=element_text(size=10),
        plot.margin = margin(4.5,4,4.5,4, "cm"))+
  labs(x="Log2 Fold Change",y="-log10 (adj.p)")
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_DOD/WithGeneSymbol_Result_RUV2.txt",row.names = 1)

library(ggrepel)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_DOD/Volcano_T0_T1.pdf")
ggplot(data, aes(x=log2FoldChange, y=-log10(padj))) +
  geom_point(data=subset(data, padj <.05 & log2FoldChange <= -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#166b99",fill="#26b3ff",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange >= 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,fill="#ffad73",color="#996745",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange > 0 & log2FoldChange < 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#ffcdab",fill="#ffe6d5",size=1)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange < 0 & log2FoldChange > -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#7cd1ff",fill="#bde8ff",size=1)+
  geom_point(data=subset(data, padj >=.05),aes(x=log2FoldChange,y=-log10(padj)),pch=21,color="#8c8c8c",fill="#cccccc",size=0.5)+
  geom_vline(xintercept=1, linetype="dashed",size=0.35)+
  geom_vline(xintercept=-1, linetype="dashed",size=0.35)+scale_x_continuous(limits = c(-15, 12), breaks = seq(-15, 12, by = 3))+
  geom_hline(yintercept=1.3010299957, linetype="dashed",size=0.35)+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2,label.size = NA,segment.alpha=0.5,
                   box.padding=0.8,nudge_x = 0.1,nudge_y = 0.1,max.overlaps = 2000)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),
        axis.title=element_text(size=10),axis.text.y=element_text(size=10),axis.text.x=element_text(size=10),
        plot.margin = margin(4.5,4,4.5,4, "cm"))+
  labs(x="Log2 Fold Change",y="-log10 (adj.p)")
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/WithGeneSymbol_Result_RUV3.txt",row.names = 1)

library(ggrepel)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Volcano_T2_T0.pdf")
ggplot(data, aes(x=log2FoldChange, y=-log10(padj))) +
  geom_point(data=subset(data, padj <.05 & log2FoldChange <= -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#166b99",fill="#26b3ff",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange >= 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,fill="#ffad73",color="#996745",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange > 0 & log2FoldChange < 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#ffcdab",fill="#ffe6d5",size=1)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange < 0 & log2FoldChange > -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#7cd1ff",fill="#bde8ff",size=1)+
  geom_point(data=subset(data, padj >=.05),aes(x=log2FoldChange,y=-log10(padj)),pch=21,color="#8c8c8c",fill="#cccccc",size=0.5)+
  geom_vline(xintercept=1, linetype="dashed",size=0.35)+
  geom_vline(xintercept=-1, linetype="dashed",size=0.35)+scale_x_continuous(limits = c(-10, 8), breaks = seq(-10, 8, by = 3))+
  geom_hline(yintercept=1.3010299957, linetype="dashed",size=0.35)+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2,label.size = NA,segment.alpha=0.5,
                   box.padding=0.8,nudge_x = 0.1,nudge_y = 0.1,max.overlaps = 2000)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),
        axis.title=element_text(size=10),axis.text.y=element_text(size=10),axis.text.x=element_text(size=10),
        plot.margin = margin(4.5,4,4.5,4, "cm"))+
  labs(x="Log2 Fold Change",y="-log10 (adj.p)")
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOD_vs_OneMonth/WithGeneSymbol_Result_RUV2.txt",row.names = 1)

library(ggrepel)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOD_vs_OneMonth/Volcano_T1_T2.pdf")
ggplot(data, aes(x=log2FoldChange, y=-log10(padj))) +
  geom_point(data=subset(data, padj <.05 & log2FoldChange <= -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#166b99",fill="#26b3ff",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange >= 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,fill="#ffad73",color="#996745",size=1.5)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange > 0 & log2FoldChange < 1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#ffcdab",fill="#ffe6d5",size=1)+
  geom_point(data=subset(data, padj <.05 & log2FoldChange < 0 & log2FoldChange > -1),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#7cd1ff",fill="#bde8ff",size=1)+
  geom_point(data=subset(data, padj >=.05),aes(x=log2FoldChange,y=-log10(padj)),pch=21,color="#8c8c8c",fill="#cccccc",size=0.5)+
  geom_vline(xintercept=1, linetype="dashed",size=0.35)+
  geom_vline(xintercept=-1, linetype="dashed",size=0.35)+scale_x_continuous(limits = c(-20, 5), breaks = seq(-20, 5, by = 3))+
  geom_hline(yintercept=1.3010299957, linetype="dashed",size=0.35)+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2,label.size = NA,segment.alpha=0.5,
                   box.padding=0.8,nudge_x = 0.1,nudge_y = 0.1,max.overlaps = 2000)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),
        axis.title=element_text(size=10),axis.text.y=element_text(size=10),axis.text.x=element_text(size=10),
        plot.margin = margin(4.5,4,4.5,4, "cm"))+
  labs(x="Log2 Fold Change",y="-log10 (adj.p)")
dev.off()
```

```{r}
library(reshape)
R=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Betweenness/Top20_Corr.txt",header = TRUE)
head(R)
Cas=cast(R)
head(Cas)
write.table(t(Cas),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Betweenness/Top20_CorrCast.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Betweenness/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Cohort=c("T0"="#a70000","T1"="#e4b2b2","T2"="#738957")))


FBA=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Betweenness/Top20_CorrCast.txt",row.names = 1)


col_tpm = colorRamp2(c(4,2,1,0.5, 0,-0.5,-1,-2,-4), c("#333300","#666600","#e5e500","#ffffcc","white","#d8b2d8","#a64ca6","#730073","#400040"))

#col_tpm2 = colorRamp2(c(1.5,0.75,0.5,0,-0.5,-0.75,-1.5), c("#ffff19","#99990f","#4c4c07","black","#003333","#00b2b2","#00ffff"))
col_tpm2 = colorRamp2(c(1.5,0.75,0.5,0,-0.5,-0.75,-1.5), c("#ffff19","#99990f","#4c4c07","black","#203470","#3454b4","#4169e1"))

col_corr = colorRamp2(c(1,0.75,0.5,0.25, 0,-0.25,-0.5,-0.75,-1), c("#ff0000","#ff3232","#ff7f7f","#ffcccc","white","#e5e5ff",
"#7f7fff","#3232ff","#0000ff"))
gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Betweenness/GeneInfo.txt",row.names = 1)
nrow(gene)
gene_anno2 = HeatmapAnnotation(df = gene, simple_anno_size = unit(0.2, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Comm=c("C1"="#997597",
                                                "C2"="#b2c3df","C3"="#6d6f76","C4"="#9e6b34","C5"="#d7aa7a","C6"="#e9cae7")))


H1=Heatmap(as.matrix((FBA)),col=col_corr,cluster_rows=FALSE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),bottom_annotation = gene_anno2,column_split = gene$Comm,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1,0.5,0,-0.5,-1),
                                       labels_gp = gpar(fontsize = 8)),
           name = "spearman",show_row_names = FALSE,row_names_gp=gpar(fontsize = 6),
           height  = unit(11, "cm"),width  = unit(11, "cm"))

TPM=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Betweenness/Top20_TPM.txt",row.names = 1)

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Betweenness/GeneInfo.txt",row.names = 1)
nrow(gene)
gene_anno = rowAnnotation(df = gene, simple_anno_size = unit(0.2, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Comm=c("C1"="#997597",
                                                "C2"="#b2c3df","C3"="#6d6f76","C4"="#9e6b34","C5"="#d7aa7a","C6"="#e9cae7")))
nrow(as.matrix(t(scale(t(TPM)))))
H2=Heatmap(as.matrix(t(scale(t(TPM)))),col=col_tpm2,cluster_rows=FALSE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),column_split = chrt$Cohort,left_annotation = gene_anno,top_annotation = chrt_anno,row_split = gene$Comm,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),at = c(2,1,0,-1,-2),
                                       title_gp = gpar(fontsize = 8),
                                       labels_gp = gpar(fontsize = 8)),
           name = "z-score(TPM)",show_row_names = TRUE,row_names_gp=gpar(fontsize = 6),
           height  = unit(11, "cm"),width  = unit(11, "cm"))

tt=H2+H1
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Betweenness/Top20_Corr.pdf",width = 10,height = 10)
draw(tt, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Heatmap/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Cohort=c("T0"="#a70000","T1"="#e4b2b2","T2"="#738957")))


FBA=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Heatmap/Top20_CorrCst.txt",row.names = 1)


col_tpm = colorRamp2(c(4,2,1,0.5, 0,-0.5,-1,-2,-4), c("#333300","#666600","#e5e500","#ffffcc","white","#d8b2d8","#a64ca6","#730073","#400040"))

col_tpm2 = colorRamp2(c(2,1,0.5,0,-0.5,-1,2), c("#cccc00","#e5e500","#ffff19","#00ffff","#00b2b2","#001919","black"))

col_corr = colorRamp2(c(1,0.75,0.5,0.25, 0,-0.25,-0.5,-0.75,-1), c("#ff0000","#ff3232","#ff7f7f","#ffcccc","white","#e5e5ff",
"#7f7fff","#3232ff","#0000ff"))
gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Heatmap/GeneInfo.txt",row.names = 1)
nrow(gene)
gene_anno2 = HeatmapAnnotation(df = gene, simple_anno_size = unit(0.2, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Comm=c("C1"="#dba8d8",
                                                "C2"="#b2c3df","C3"="#3b5998","C4"="#c68642","C5"="#b5a4d0","C6"="#6d6f76")))


H1=Heatmap(as.matrix((FBA)),col=col_corr,cluster_rows=FALSE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),bottom_annotation = gene_anno2,column_split = gene$Comm,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1,0.5,0,-0.5,-1),
                                       labels_gp = gpar(fontsize = 8)),
           name = "spearman",show_row_names = FALSE,row_names_gp=gpar(fontsize = 6),
           height  = unit(11, "cm"),width  = unit(11, "cm"))

TPM=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Heatmap/Top20_TPM.txt",row.names = 1)

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Heatmap/GeneInfo.txt",row.names = 1)
nrow(gene)
gene_anno = rowAnnotation(df = gene, simple_anno_size = unit(0.2, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Comm=c("C1"="#dba8d8",
                                                "C2"="#b2c3df","C3"="#3b5998","C4"="#c68642","C5"="#b5a4d0","C6"="#6d6f76")))
nrow(as.matrix(t(scale(t(TPM)))))
H2=Heatmap(as.matrix(t(scale(t(TPM)))),col=col_tpm2,cluster_rows=FALSE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),column_split = chrt$Cohort,left_annotation = gene_anno,top_annotation = chrt_anno,row_split = gene$Comm,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),at = c(2,1,0,-1,-2),
                                       title_gp = gpar(fontsize = 8),
                                       labels_gp = gpar(fontsize = 8)),
           name = "z-score(TPM)",show_row_names = TRUE,row_names_gp=gpar(fontsize = 6),
           height  = unit(11, "cm"),width  = unit(11, "cm"))

tt=H2+H1
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/NetworkAnalysis/Global/results/Corr_1e10/Splitted/Heatmap/Top20_Corr.pdf",width = 10,height = 10)
draw(tt, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```

```{r}
library(TMixClust)
data(toy_data_df)
print(head(toy_data_df))
plot_time_series_df(toy_data_df)
```

```{r}
library(TMixClust)
tpm=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/tmp.txt",row.names = 1)
cluster_obj = TMixClust(tpm, nb_clusters = 4,em_iter_max = 100, mc_em_iter_max = 10)
?TMixClust
generate_TMixClust_report(cluster_obj, report_folder = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/TPM", data_color = "#fd8d3c", x_label = "TOI",y_label = "value")

tpmS=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/TPM_Scaled_gene.txt",row.names = 1)
cluster_objS = TMixClust(tpmS, nb_clusters = 4)

generate_TMixClust_report(cluster_objS, report_folder = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/Scaled/", data_color = "#fd8d3c", x_label = "TOI",y_label = "value")

```


```{r,eval=FALSE}

## LRT

library(tximport)
library(tximportData)
library(DESeq2)
library(RUVSeq)
library(dplyr)
library(NOISeqNOISeq)
design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/LRT/Design.txt",header = TRUE)
files <- file.path("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq",design$ID, "CodingCount.txt")
names(files) <- design$ID
txi <- tximport(files, type="salmon",txIn=FALSE,geneIdCol="Name")

count=as.data.frame(txi$counts)
count=round(count, 0)
mytmm = tmm(count)
write.table(as.data.frame(mytmm),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/LRT/TMM.txt",sep="\t", quote=FALSE,col.names = NA)

write.table(as.data.frame(count),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/LRT/Count.txt",sep="\t", quote=FALSE,col.names = NA)


Design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/LRT/Design.txt",row.names = 1, check.names = FALSE)
filtered=count%>%filter(rowSums(across(where(is.numeric)))!=0)
Myset <- newSeqExpressionSet(as.matrix(filtered),phenoData = data.frame(Design))
Myset <- betweenLaneNormalization(Myset, which="upper")
design <- model.matrix(~Time+Age+Gender+plasmacytoid_DC+otherCells+intermediate_monocyte+non_classical_monocyte+naive_CD4_Tcell+neutrophil+memory_CD4_Tcell+classical_monocyte+myeloid_DC+Nkcell+MAIT_Tcell, data=pData(Myset))
genes <- rownames(filtered)[grep("^ENS", rownames(filtered))]

y <- DGEList(counts=counts(Myset), group=as.factor(Design$Time))
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
res <- residuals(fit, type="deviance")
Myset2 <- RUVr(Myset, genes, k=1, res)

ddsTxi <- DESeqDataSetFromTximport(txi,colData = pData(Myset2),design = ~ W_1 + Time)
dds=DESeq(ddsTxi,test="LRT", reduced=~W_1)

out=getVarianceStabilizedData(dds)
write.table(as.data.frame(out),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/LRT/VST.txt",sep="\t", quote=FALSE,col.names = NA)
res=results(dds,independentFiltering = FALSE)                       
write.table(as.data.frame(res),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/LRT/Result_RUV.txt",sep="\t", quote=FALSE,col.names = NA)

```

```{r,eval=FALSE}
library(PCAtools)
count=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2.txt",row.names = 1,check.names = FALSE)
meta=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/Info.txt",row.names = 1)
p <- PCAtools::pca((count), metadata = meta, removeVar = 0.1)
write.table(p$rotated,file = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2_PCA.txt",sep = "\t",col.names = NA,quote = FALSE)
head(p$variance)

```

```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2_PCA.txt",row.names = 1,check.names = FALSE)
gg <- merge(dat,aggregate(cbind(mean.pc1=PC1,mean.pc2=PC2)~Cohort,dat,mean),by="Cohort")
head(gg)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2_PCA.pdf")
ggplot(gg, aes(PC1,PC2,color=Cohort))+geom_point(size=3,aes(fill=Cohort),shape=21)+
  geom_point(aes(x=mean.pc1,y=mean.pc2,fill=Cohort,color=Cohort),size=3.5)+theme_gray()+
  geom_segment(aes(x=mean.pc1, y=mean.pc2, xend=PC1, yend=PC2))+
    scale_color_manual(values=c("Healthy"="#486129","T2"="#315b7d"))+
    scale_fill_manual(values=c("Healthy"="#738957","T2"="#4682B4"))+
  labs(x="PC1, 97.72% variation",y="PC2, 1.33% variation")+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = Cohort), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/pwy.txt",header = TRUE,check.names = FALSE,row.names = 1)
head(data)

col_fun1 = colorRamp2(c(0,0.1,0.5,1,1.5,1.75,2,2.5), c("white" ,"#b4dce4","#6abac9","#2098ae","#088da5","#067084","#045463","#033842"))

sf=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/type.txt",header = TRUE,row.names = 1)
ha = HeatmapAnnotation(df = sf,show_annotation_name = FALSE,annotation_name_side = "right",simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.3, "cm"),
                                                      grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8), 
                                                      labels_gp = gpar(fontsize = 8)),
                       col = list(Type=c("a) p adj (dist.dir.dn)"="#26b3ff","b) p adj (mix.dir.dn)"="#bde8ff","c) p adj (non-dir.)"="grey","d) p adj (mix.dir.up)"="#ffe6d5","e) p adj (dist.dir.up)"="#ffad73")))


H1=Heatmap(as.matrix(-log10(data)),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,
           row_title_gp = gpar(fontsize=0),row_dend_width = unit(0.5, "cm"),
           column_title_gp =gpar(fontsize = 0),top_annotation = ha,
           heatmap_legend_param =list(grid_width = unit(0.3, "cm"),grid_height = unit(0.3, "cm"),title_gp = gpar(fontsize = 8),
                                      labels_gp = gpar(fontsize = 8)),
           name = "-log10(Padj)",show_row_names = TRUE,row_names_gp=gpar(fontsize = 7),height  = unit(10, "cm"),
           width  = unit(2, "cm"),column_split = sf$Comparison)


pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Healthy_vs_OneMonth/pwy.pdf")
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "right",merge_legend = TRUE)
dev.off()
```

```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/heatmap/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.15, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.3, "cm"),
                                                              grid_height = unit(0.3, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Cohort=c("1. HC"="#4682B4","2. T0"="#a70000","3. T1"="#e4b2b2","4. T2"="#738957")))

Eu=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/heatmap/TPM_gene.txt",row.names = 1)
#col_fun1 = colorRamp2(c(1.5,0.75,0.5,0,-0.5,-0.75,-1.5), c("#ffff19","#99990f","#4c4c07","black","#003333","#00b2b2","#00ffff"))

col_fun1 = colorRamp2(c(1.5,0.75,0.5,0,-0.5,-0.75,-1.5), c("#ffff19","#99990f","#4c4c07","black","#203470","#3454b4","#4169e1"))

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/heatmap/GeneInfo.txt",row.names = 1)
gene_anno = rowAnnotation(df = gene, simple_anno_size = unit(0.15, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.3, "cm"),
                                                              grid_height = unit(0.3, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Cluster=c("c1"="#b7bdd8","c2"="#bdd8b7","c3"="#b7d8d2","c4"="#ffebaf")))

H1=Heatmap(as.matrix(t(scale(t(Eu)))),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,use_raster = TRUE,
           na_col = "white",
           row_title_gp =gpar(fontsize = 0),row_split = gene$Cluster,left_annotation = gene_anno,
           row_dend_width = unit(0.75, "cm"),column_title_gp =gpar(fontsize = 0),top_annotation = chrt_anno,column_split = chrt$Cohort,
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.3, "cm"),grid_height = unit(0.3, "cm"),
                                       title_gp = gpar(fontsize = 8),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Z-score(TPM)",show_row_names =FALSE,row_names_gp=gpar(fontsize =8),
           height  = unit(10, "cm"),width  = unit(8, "cm"))

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/heatmap/TPM_gene.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right",)
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/heatmap/pwy.txt")
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/heatmap/pwy.pdf")
ggplot(data) +
  geom_point(aes(x=1,y=factor(Term,levels = unique(Term)),size=Overlap,color=pval))+scale_x_discrete(limits=c("1"))+
  scale_color_gradient(low="#CC9999",high="#800000")+theme_bw()+
  theme(axis.title.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank(),axis.text.y = element_text(size=8,color="black"),plot.title = element_text(hjust = 0.5),
        axis.ticks = element_blank(),legend.text = element_text(colour = "black"),plot.margin = margin(4,5,5,3.5, "cm"),
        legend.direction = "vertical",
        legend.title = element_text(size=8),panel.border = element_blank(),panel.grid.major = element_blank())+
  guides(size=guide_legend(override.aes=list(colour="grey"),title = "Gene ratio(%)",ncol = 1),
         color=guide_legend(title = "-log10(padj)",ncol = 1,override.aes = list(size = 3)))
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/clusters/C1/ip.txt")
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/clusters/C1/ip.pdf")
ggplot(data) +
  geom_point(aes(x=1,y=factor(Term,levels = unique(Term)),size=Overlap,color=pval))+scale_x_discrete(limits=c("1"))+
  scale_color_gradient(low="#9cd1db",high="#056273")+theme_bw()+
  theme(axis.title.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank(),axis.text.y = element_text(size=8,color="black"),plot.title = element_text(hjust = 0.5),
        axis.ticks = element_blank(),legend.text = element_text(colour = "black"),plot.margin = margin(8,6,8,5, "cm"),
        legend.direction = "vertical",
        legend.title = element_text(size=8),panel.border = element_blank(),panel.grid.major = element_blank())+
  guides(size=guide_legend(override.aes=list(colour="grey"),title = "Gene ratio(%)",ncol = 1),
         color=guide_legend(title = "-log10(padj)",ncol = 1,override.aes = list(size = 3)))
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/clusters/C3/ip.txt")
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/clusters/C3/ip.pdf")
ggplot(data) +
  geom_point(aes(x=1,y=factor(Term,levels = unique(Term)),size=Overlap,color=pval))+scale_x_discrete(limits=c("1"))+
  scale_color_gradient(low="#9cd1db",high="#056273")+theme_bw()+
  theme(axis.title.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank(),axis.text.y = element_text(size=8,color="black"),plot.title = element_text(hjust = 0.5),
        axis.ticks = element_blank(),legend.text = element_text(colour = "black"),plot.margin = margin(4,3,4,5, "cm"),
        legend.direction = "vertical",
        legend.title = element_text(size=8),panel.border = element_blank(),panel.grid.major = element_blank())+
  guides(size=guide_legend(override.aes=list(colour="grey"),title = "Gene ratio(%)",ncol = 1),
         color=guide_legend(title = "-log10(padj)",ncol = 1,override.aes = list(size = 3)))
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/clusters/C4/ip.txt")
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/clusters/C4/ip.pdf")
ggplot(data) +
  geom_point(aes(x=1,y=factor(Term,levels = unique(Term)),size=Overlap,color=pval))+scale_x_discrete(limits=c("1"))+
  scale_color_gradient(low="#9cd1db",high="#056273")+theme_bw()+
  theme(axis.title.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank(),axis.text.y = element_text(size=8,color="black"),plot.title = element_text(hjust = 0.5),
        axis.ticks = element_blank(),legend.text = element_text(colour = "black"),plot.margin = margin(6,4.3,6,5, "cm"),
        legend.direction = "vertical",
        legend.title = element_text(size=8),panel.border = element_blank(),panel.grid.major = element_blank())+
  guides(size=guide_legend(override.aes=list(colour="grey"),title = "Gene ratio(%)",ncol = 1),
         color=guide_legend(title = "-log10(padj)",ncol = 1,override.aes = list(size = 3)))
dev.off()
```

```{r,fig.align = "center",fig.height=13, fig.width=10, echo=FALSE,fig.cap ="Cell type proportions computed. Pvalues corresponding to Mann-Whiteny test are labelled",warning=FALSE,message=FALSE}
M=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/MIsc/Ip.txt",header = TRUE,check.names = FALSE)
library(reshape2)

head(M)
library(ggpubr)
library(ggplot2)

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/MIsc/Ip.pdf")
ggplot(M,aes(x=factor(Cluster,levels = unique(Cluster)),y=OneYr,color=Cluster,fill=Cluster))+geom_boxplot(outlier.shape = NA)+
  facet_wrap(~ Cluster, ncol = 4,scales ="free")+theme_bw()+
  geom_jitter(shape=16, size=0.25,color="#808080",position=position_jitter(0.05))+
scale_color_manual(values=c("cluster1"="#808497","cluster2"="#71816d",
                                "cluster3"="#809793","cluster4"="#b2a47a"))+
    scale_fill_manual(values=c("cluster1"="#b7bdd8","cluster2"="#bdd8b7","cluster3"="#b7d8d2","cluster4"="#ffebaf"))+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank(),legend.position ="none",panel.border = element_blank(),
        axis.ticks.x = element_blank(),axis.text.x = element_blank(),legend.title = element_blank(),legend.text = element_text(size = 8),
        plot.margin = margin(7,5.5,7,5.5, "cm"),strip.text.x = element_text(size = 8),
        axis.text.y = element_text(color="black",size=8),plot.title = element_text(hjust = 0.5,size = 8))+
  guides(color = guide_legend(nrow = 1),fill = guide_legend(nrow = 1))
dev.off()
```


```{r,fig.align = "center",fig.height=13, fig.width=10, echo=FALSE,fig.cap ="Cell type proportions computed. Pvalues corresponding to Mann-Whiteny test are labelled",warning=FALSE,message=FALSE}
M=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/hc.txt",header = TRUE,check.names = FALSE)
library(reshape2)

head(M)
library(ggpubr)
library(ggplot2)

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/HC.pdf")
ggplot(M,aes(x=factor(Cluster,levels = unique(Cluster)),y=HC,color=Cluster,fill=Cluster))+geom_boxplot(outlier.shape = NA)+
  facet_wrap(~ Cluster, ncol = 4,scales ="free")+theme_bw()+
  geom_jitter(shape=16, size=0.25,color="#808080",position=position_jitter(0.05))+
scale_color_manual(values=c("cluster1"="#808497","cluster2"="#71816d",
                                "cluster3"="#809793","cluster4"="#b2a47a"))+
    scale_fill_manual(values=c("cluster1"="#b7bdd8","cluster2"="#bdd8b7","cluster3"="#b7d8d2","cluster4"="#ffebaf"))+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank(),legend.position ="none",panel.border = element_blank(),
        axis.ticks.x = element_blank(),axis.text.x = element_blank(),legend.title = element_blank(),legend.text = element_text(size = 8),
        plot.margin = margin(7,5.5,7,5.5, "cm"),strip.text.x = element_text(size = 8),
        axis.text.y = element_text(color="black",size=8),plot.title = element_text(hjust = 0.5,size = 8))+
  guides(color = guide_legend(nrow = 1),fill = guide_legend(nrow = 1))
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Volcano_LRT.txt",row.names = 1)

library(ggrepel)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Volcano_LRT.pdf")
ggplot(data, aes(x=log2FoldChange, y=-log10(padj))) +
  geom_point(data=subset(data, LRT == "No" ),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#595959",fill="#999999",size=1.5)+
  geom_point(data=subset(data, LRT == "Yes" ),aes(x=log2FoldChange,y=-log10(padj)),
             pch=21,color="#b27300",fill="#ffa500",size=1.5)+
  geom_vline(xintercept=1, linetype="dashed",size=0.35)+
  geom_vline(xintercept=-1, linetype="dashed",size=0.35)+scale_x_continuous(limits = c(-10, 8), breaks = seq(-10, 8, by = 3))+
  geom_hline(yintercept=1.3010299957, linetype="dashed",size=0.35)+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2,label.size = NA,segment.alpha=0.5,
                   box.padding=0.8,nudge_x = 0.1,nudge_y = 0.1,max.overlaps = 2000)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),
        axis.title=element_text(size=10),axis.text.y=element_text(size=10),axis.text.x=element_text(size=10),
        plot.margin = margin(4.5,4,4.5,4, "cm"))+
  labs(x="Log2 Fold Change",y="-log10 (adj.p)")
dev.off()
```


```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/Volcano_LRT.txt",row.names = 1)

library(ggrepel)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/DOH_vs_OneMonth/MA_LRT.pdf")
ggplot(data, aes(x=log2(baseMean),y=log2FoldChange)) +
  geom_point(data=subset(data, LRT == "No" ),aes(x=log2(baseMean),y=log2FoldChange),
             pch=21,color="#8c8c8c",fill="#cccccc",size=1.5)+
  geom_point(data=subset(data, LRT == "Yes" & padj < 0.05 ),aes(x=log2(baseMean),y=log2FoldChange),
             pch=21,color="#b27300",fill="#ffa500",size=1.5)+
  geom_point(data=subset(data, LRT == "Yes" & padj >= 0.05 ),aes(x=log2(baseMean),y=log2FoldChange),
             pch=1,color="#996300",size=1.5,stroke = 0.75)+
  geom_hline(yintercept=1, linetype="dashed",size=0.35)+
  geom_hline(yintercept=-1, linetype="dashed",size=0.35)+
  geom_label_repel(aes(label = Label),segment.color = '#cccccc',fill=NA,color="black",size=2,label.size = NA,segment.alpha=0.5,
                   box.padding=0.8,nudge_x = 0.1,nudge_y = 0.1,max.overlaps = 2000)+
  theme(legend.title=element_text(size=8),legend.text=element_text(size=6),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),
        axis.title=element_text(size=10),axis.text.y=element_text(size=10),axis.text.x=element_text(size=10),
        plot.margin = margin(4.5,4,4.5,4, "cm"))+
  labs(x="log2(BaseMean)",y="log2FoldChange")
dev.off()
```

```{r}
library(reshape)
R=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/Associaition/Heatmap/All/Selected_Corr.txt",header = TRUE)
head(R)
Cas=cast(R)
head(Cas)
write.table((Cas),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/Associaition/Heatmap/All/Selected_CorrCasted.txt",sep="\t",row.names =FALSE,quote = FALSE,na = "0")

dim(Cas)
```

```{r}
library(circlize)
library(ComplexHeatmap)

FBA=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/Associaition/Heatmap/All/Selected_CorrCasted.txt",row.names = 1)

col_corr = colorRamp2(c(1,0.95,0.9,0.75,0.5, 0,-0.5,-0.75,-0.9,-0.95,-1), c("#b20000","#cc0000","#e50000","#ff0000",
"#ff7f7f","white","#7f7fff","#0000ff",
"#0000e5","#0000cc","#0000b2"))

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/Associaition/Heatmap/All/Gene2.txt",row.names = 1)
nrow(gene)
gene_anno2 = HeatmapAnnotation(df = gene, simple_anno_size = unit(0.2, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Module=c("c1"="#B7BDD8","c2"="#BDD8B7","c3"="#B7D8D2","c4"="#FFEBAF")))

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/Associaition/Heatmap/All/Gene2.txt",row.names = 1)
gene_anno = rowAnnotation(df = gene, simple_anno_size = unit(0.2, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Module=c("c1"="#B7BDD8","c2"="#BDD8B7","c3"="#B7D8D2","c4"="#FFEBAF")))

H1=Heatmap(as.matrix((FBA)),col=col_corr,cluster_rows=FALSE,cluster_columns = FALSE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),top_annotation = gene_anno2,column_split = gene$Module,row_split = gene$Module,
           row_dend_width = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),left_annotation = gene_anno,
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1,0.5,0,-0.5,-1),
                                       labels_gp = gpar(fontsize = 8)),
           name = "spearman",show_row_names = FALSE,row_names_gp=gpar(fontsize = 6),
           height  = unit(7, "cm"),width  = unit(7, "cm"))

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/Associaition/Heatmap/All/Heatmap.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r,eval=FALSE}
library(ComplexHeatmap)
library(circlize)
CelFr=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/Sub/SubSytem.txt",header = TRUE,check.names = FALSE,row.names = 1)



sample=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/SampleInfo.txt",row.names = 1)

ha = HeatmapAnnotation(df = sample,show_annotation_name = FALSE,simple_anno_size = unit(0.3, "cm"),
                       annotation_legend_param = list(Group = list(direction = "horizontal",grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10), 
                                                                   labels_gp = gpar(fontsize = 10))),
                       col = list(Group=c("1. HC"="#4682B4","2. T0"="#a70000","3. T1"="#e4b2b2","4. T2"="#738957")))

col_fun1 = colorRamp2(c(1000,750,500, 0,-500,-750,-1000), c("#7F7F00","#B2B200" ,"#E5E500","white","#BF7FBF","#993299","#590059"))
col_fun1 = colorRamp2(c(1000,500,150,0,-150,-500,-1000), c("#ffff00","#cccc00","#7f7f00","white","#999999","#4c4c4c","#000000"))
col_fun1 = colorRamp2(c(1000,0,-1000), c("#ffff00","black","#0000ff"))

info2=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/Sub/SubSytemInfo.txt",row.names = 1)
ha2 = rowAnnotation(df = info2,show_annotation_name = FALSE,simple_anno_size = unit(0.3, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.5, "cm"),
                                                      grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10), 
                                                                   labels_gp = gpar(fontsize = 10)),
                       col = list(SubSystem=c("Amino Acid Metabolism"="#b8f4bc",
                       "Glycolysis / Gluconeogenesis"="#94967b","Other"="#bfbfbf","Oxidative phosphorylation"="#eecff4","Pentose phosphate pathway"="#cecdff","Pyruvate metabolism"="#f2b894","TCA cycle"="#97f4e9")))


ha3 = rowAnnotation(foo = anno_mark(at = c(37,38,39,40,41,42,43,44,45,46,47,48,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315),
                                   labels_gp = gpar(fontsize=10),lines_gp = gpar(col="#8e8e8e"),link_width=unit(8, "mm"),labels = c("CoA[m] + NAD+[m] + pyruvate[m] => acetyl-CoA[m] + CO2[m] + NADH[m]","H+[p] + NADH[p] + pyruvate[p] <=> L-lactate[p] + NAD+[p]","acetaldehyde[c] + H2O[c] + NADP+[c] => acetate[c] + 2 H+[c] + NADPH[c]","DHAP[c] + erythrose-4-phosphate[c] => sedoheptulose-1,7-bisphosphate[c]","DHAP[c] + GAP[c] <=> fructose-1,6-bisphosphate[c]","ATP[c] + fructose-6-phosphate[c] => ADP[c] + fructose-1,6-bisphosphate[c] + H+[c]","DHAP[c] <=> GAP[c]","ATP[c] + glucose[c] => ADP[c] + glucose-6-phosphate[c] + H+[c]","glucose-1-phosphate[c] <=> glucose-6-phosphate[c]","beta-D-glucose[c] <=> glucose[c]","ATP[c] + beta-D-glucose[c] => ADP[c] + beta-D-glucose-6-phosphate[c] + H+[c]","acetaldehyde[m] + H2O[m] + NAD+[m] => acetate[m] + 2 H+[m] + NADH[m]","4 ferrocytochrome C[m] + 7.92 H+[m] + O2[m] => 4 ferricytochrome C[m] + 1.96 H2O[m] + 0.02 O2-[m] + 4 H+[i]","FADH2[m] + ubiquinone[m] <=> FAD[m] + ubiquinol[m]","ADP[m] + Pi[m] + 3 H+[i] => ATP[m] + 2 H+[m] + H2O[m]","2 ferricytochrome C[m] + 2 H+[m] + ubiquinol[m] => 2 ferrocytochrome C[m] + ubiquinone[m] + 4 H+[i]","3 glucose-6-phosphate[c] + 3 NADP+[c] => 3 glucono-1,5-lactone-6-phosphate[c] + 3 H+[c] + 3 NADPH[c]","glucono-1,5-lactone-6-phosphate[c] + H+[c] + NADPH[c] <=> glucose-6-phosphate[c] + NADP+[c]","ribose-5-phosphate[c] <=> ribulose-5-phosphate[c]","ribose-1-phosphate[c] <=> ribose-5-phosphate[c]","2-deoxy-D-ribose-5-phosphate[c] <=> acetaldehyde[c] + GAP[c]","D-xylulose-5-phosphate[c] + erythrose-4-phosphate[c] <=> fructose-6-phosphate[c] + GAP[c]","D-xylulose-5-phosphate[c] + ribose-5-phosphate[c] <=> GAP[c] + sedoheptulose-7-phosphate[c]","GAP[c] + sedoheptulose-7-phosphate[c] <=> erythrose-4-phosphate[c] + fructose-6-phosphate[c]","ADP[c] + H+[c] + sedoheptulose-1,7-bisphosphate[c] <=> ATP[c] + sedoheptulose-7-phosphate[c]","H+[c] + sedoheptulose-1,7-bisphosphate[c] + UDP[c] <=> sedoheptulose-7-phosphate[c] + UTP[c]","2-deoxy-D-ribose-1-phosphate[c] <=> 2-deoxy-D-ribose-5-phosphate[c]","acetaldehyde[c] + H2O[c] + NAD+[c] => acetate[c] + 2 H+[c] + NADH[c]","ATP[m] + HCO3-[m] + pyruvate[m] => ADP[m] + H+[m] + OAA[m] + Pi[m]","CDP[c] + H+[c] + PEP[c] => CTP[c] + pyruvate[c]","H+[m] + NADH[m] + pyruvate[m] <=> L-lactate[m] + NAD+[m]","acetoacetate[m] + succinyl-CoA[m] => acetoacetyl-CoA[m] + succinate[m]","H+[m] + oxalosuccinate[m] => AKG[m] + CO2[m]","H+[c] + NADH[c] + OAA[c] <=> malate[c] + NAD+[c]","H+[m] + NADH[m] + OAA[m] <=> malate[m] + NAD+[m]","citrate[m] + CoA[m] + H+[m] <=> acetyl-CoA[m] + H2O[m] + OAA[m]","citrate[c] <=> isocitrate[c]","citrate[m] <=> isocitrate[m]","cis-aconitate[m] + H2O[m] <=> citrate[m]","H+[m] + NADPH[m] + oxalosuccinate[m] <=> isocitrate[m] + NADP+[m]","H+[m] + NADH[m] + oxalosuccinate[m] <=> isocitrate[m] + NAD+[m]","cis-aconitate[m] + H2O[m] <=> isocitrate[m]","fumarate[m] + ubiquinol[m] <=> succinate[m] + ubiquinone[m]","AKG[m] + CoA[m] + NAD+[m] => CO2[m] + NADH[m] + succinyl-CoA[m]","FADH2[m] + fumarate[m] <=> FAD[m] + succinate[m]")))

H1=Heatmap(CelFr,cluster_columns = FALSE,show_column_names = FALSE,col = col_fun1,top_annotation = ha,row_dend_width = unit(0.5, "cm"),
           ,row_gap = unit(0.5, "mm"),left_annotation = ha2,row_split = info2$SubSystem,row_dend_reorder = TRUE,
           column_dend_height = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),cluster_rows = TRUE,row_title_gp = gpar(fontsize = 0),
           heatmap_legend_param =list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10),
                                      labels_gp = gpar(fontsize = 10)),
           name = "Flux",show_row_names = FALSE,row_names_gp=gpar(fontsize = 12),height  = unit(18, "cm"),width  = unit(5, "cm"))


pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/Sub/HeatMap.pdf",width = 12,height = 12)
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "right",merge_legend = TRUE)
dev.off()
```

```{r,eval=FALSE}
library(ComplexHeatmap)
library(circlize)
CelFr=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/Transport.txt",header = TRUE,check.names = FALSE,row.names = 1)



sample=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/SampleInfo.txt",row.names = 1)

ha = HeatmapAnnotation(df = sample,show_annotation_name = FALSE,simple_anno_size = unit(0.3, "cm"),
                       annotation_legend_param = list(Group = list(direction = "horizontal",grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10), 
                                                                   labels_gp = gpar(fontsize = 10))),
                       col = list(Group=c("1. HC"="#4682B4","2. T0"="#a70000","3. T1"="#e4b2b2","4. T2"="#738957")))

col_fun1 = colorRamp2(c(1000,0,-1000), c("#ffff00","black","#0000ff"))


info2=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/TransportInfo.txt",row.names = 1)
ha2 = rowAnnotation(df = info2,show_annotation_name = FALSE,simple_anno_size = unit(0.3, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.5, "cm"),
                                                      grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10), 
                                                                   labels_gp = gpar(fontsize = 10)),
                       col = list(SubSystem=c("Transport reactions"="#9f8170")))


H1=Heatmap(CelFr,cluster_columns = FALSE,show_column_names = FALSE,col = col_fun1,top_annotation = ha,row_dend_width = unit(0.5, "cm"),
           ,row_gap = unit(0.5, "mm"),left_annotation = ha2,row_split = info2$SubSystem,row_dend_reorder = TRUE,
           column_dend_height = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),cluster_rows = TRUE,row_title_gp = gpar(fontsize = 0),
           heatmap_legend_param =list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10),
                                      labels_gp = gpar(fontsize = 10)),
           name = "Flux",show_row_names = FALSE,row_names_gp=gpar(fontsize = 12),height  = unit(18, "cm"),width  = unit(5, "cm"))


pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/Transport.pdf",width = 12,height = 12)
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "right",merge_legend = TRUE)
dev.off()
```

```{r}
library(reshape2)
library(ggplot2)
dd=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/T0_Sub.txt")
M=melt(dd)

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/T0_Sub.pdf",width = 12,height = 12)
ggplot(M, aes(fill=variable, y=value, x=factor(EQN,levels = unique(EQN)))) + 
    geom_bar(position="dodge", stat="identity",width = 2.5)+coord_flip()+
  scale_fill_manual(values=c("HC"="#738957","T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"))+
  theme(legend.title=element_text(size=12),legend.text=element_text(size=12),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),legend.position = "bottom",
        axis.title.x=element_text(size=12),axis.title.y = element_blank(),axis.text.y=element_text(size=12),axis.text.x=element_text(size=12),
        plot.margin = margin(8.5,0.5,8,1, "cm"))
dev.off()
```

```{r}
library(reshape2)
library(ggplot2)
dd=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/T0_Trans.txt")
M=melt(dd)

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/T0_Trans.pdf",width = 12,height = 12)
ggplot(M, aes(fill=variable, y=value, x=factor(EQN,levels = unique(EQN)))) + 
    geom_bar(position="dodge", stat="identity",width = 1)+coord_flip()+
  scale_fill_manual(values=c("HC"="#738957","T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"))+
  theme(legend.title=element_text(size=12),legend.text=element_text(size=12),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),legend.position = "bottom",
        axis.title.x=element_text(size=12),axis.title.y = element_blank(),axis.text.y=element_text(size=12),axis.text.x=element_text(size=12),
        plot.margin = margin(8.5,5,8,5, "cm"))
dev.off()
```

```{r}
library(reshape2)
library(ggplot2)
dd=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/T2_Subs.txt")
M=melt(dd)

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/T2_Subs.pdf",width = 12,height = 12)
ggplot(M, aes(fill=variable, y=value, x=factor(EQN,levels = unique(EQN)))) + 
    geom_bar(position="dodge", stat="identity",width = 1)+coord_flip()+
  scale_fill_manual(values=c("HC"="#738957","T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"))+
  theme(legend.title=element_text(size=12),legend.text=element_text(size=12),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),legend.position = "bottom",
        axis.title.x=element_text(size=12),axis.title.y = element_blank(),axis.text.y=element_text(size=12),axis.text.x=element_text(size=12),
        plot.margin = margin(8.5,2,8,2, "cm"))
dev.off()
```

```{r}
library(reshape2)
library(ggplot2)
dd=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/T2_Trans.txt")
M=melt(dd)

pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/T2_Trans.pdf",width = 12,height = 12)
ggplot(M, aes(fill=variable, y=value, x=factor(EQN,levels = unique(EQN)))) + 
    geom_bar(position="dodge", stat="identity",width = 1)+coord_flip()+
  scale_fill_manual(values=c("HC"="#738957","T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"))+
  theme(legend.title=element_text(size=12),legend.text=element_text(size=12),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),legend.position = "bottom",
        axis.title.x=element_text(size=12),axis.title.y = element_blank(),axis.text.y=element_text(size=12),axis.text.x=element_text(size=12),
        plot.margin = margin(3,7,3,5, "cm"))
dev.off()
```

```{r,eval=FALSE}
library(ComplexHeatmap)
library(circlize)
CelFr=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/Sub/SubSytem.txt",header = TRUE,check.names = FALSE,row.names = 1)



sample=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/SampleInfo.txt",row.names = 1)

ha = HeatmapAnnotation(df = sample,show_annotation_name = FALSE,simple_anno_size = unit(0.3, "cm"),
                       annotation_legend_param = list(Group = list(direction = "horizontal",grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10), 
                                                                   labels_gp = gpar(fontsize = 10))),
                       col = list(Group=c("1. HC"="#4682B4","2. T0"="#a70000","3. T1"="#e4b2b2","4. T2"="#738957")))

col_fun1 = colorRamp2(c(1000,750,500, 0,-500,-750,-1000), c("#7F7F00","#B2B200" ,"#E5E500","white","#BF7FBF","#993299","#590059"))
col_fun1 = colorRamp2(c(1000,500,150,0,-150,-500,-1000), c("#ffff00","#cccc00","#7f7f00","white","#999999","#4c4c4c","#000000"))
col_fun1 = colorRamp2(c(1000,0,-1000), c("#ffff00","black","#0000ff"))

info2=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/Sub/SubSytemInfo.txt",row.names = 1)
ha2 = rowAnnotation(df = info2,show_annotation_name = FALSE,simple_anno_size = unit(0.3, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.5, "cm"),
                                                      grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10), 
                                                                   labels_gp = gpar(fontsize = 10)),
                       col = list(SubSystem=c("Amino Acid Metabolism"="#b8f4bc",
                       "Glycolysis / Gluconeogenesis"="#94967b","Other"="#bfbfbf","Oxidative phosphorylation"="#eecff4","Pentose phosphate pathway"="#cecdff","Pyruvate metabolism"="#f2b894","TCA cycle"="#97f4e9","Transport reactions"="#9f8170")))

H1=Heatmap(CelFr,cluster_columns = FALSE,show_column_names = FALSE,col = col_fun1,top_annotation = ha,row_dend_width = unit(0.5, "cm"),
           ,row_gap = unit(0.5, "mm"),left_annotation = ha2,row_split = info2$SubSystem,row_dend_reorder = TRUE,
           column_dend_height = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),cluster_rows = TRUE,row_title_gp = gpar(fontsize = 0),
           heatmap_legend_param =list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10),
                                      labels_gp = gpar(fontsize = 10)),
           name = "Flux",show_row_names = FALSE,row_names_gp=gpar(fontsize = 12),height  = unit(25, "cm"),width  = unit(5, "cm"))


pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/Sub/HeatMap.pdf",width = 12,height = 12)
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "right",merge_legend = TRUE)
dev.off()
```

```{r,eval=FALSE}
library(ComplexHeatmap)
library(circlize)
CelFr=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/personal/UniqueAll.txt",header = TRUE,check.names = FALSE,row.names = 1)



sample=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/personal/SamInfo.txt",row.names = 1)

ha = HeatmapAnnotation(df = sample,show_annotation_name = FALSE,simple_anno_size = unit(0.3, "cm"),
                       annotation_legend_param = list(Group = list(direction = "horizontal",grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10), 
                                                                   labels_gp = gpar(fontsize = 10))),
                       col = list(Group=c("1. HC"="#4682B4","2. T0"="#a70000","3. T1"="#e4b2b2","4. T2"="#738957")))

col_fun1 = colorRamp2(c(1000,750,500, 0,-500,-750,-1000), c("#7F7F00","#B2B200" ,"#E5E500","white","#BF7FBF","#993299","#590059"))
col_fun1 = colorRamp2(c(1000,500,150,0,-150,-500,-1000), c("#ffff00","#cccc00","#7f7f00","white","#999999","#4c4c4c","#000000"))
col_fun1 = colorRamp2(c(1000,0,-1000), c("#ffff00","black","#0000ff"))

info2=read.delim("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/personal/Info.txt",row.names = 1)
ha2 = rowAnnotation(df = info2,show_annotation_name = FALSE,simple_anno_size = unit(0.3, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.5, "cm"),
                                                      grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10), 
                                                                   labels_gp = gpar(fontsize = 10)),
                       col = list(SubSystem=c("Amino Acid Metabolism"="#b8f4bc",
                       "Glycolysis / Gluconeogenesis"="#94967b","Other"="#bfbfbf","Oxidative phosphorylation"="#eecff4",
                       "Pentose phosphate pathway"="#cecdff","Pyruvate metabolism"="#f2b894",
                       "TCA cycle"="#97f4e9","Transport reactions"="#9f8170")))

H1=Heatmap(CelFr,cluster_columns = FALSE,show_column_names = FALSE,col = col_fun1,top_annotation = ha,row_dend_width = unit(0.5, "cm"),
           ,row_gap = unit(0.5, "mm"),left_annotation = ha2,row_split = info2$SubSystem,row_dend_reorder = TRUE,column_split = sample$Group,
           column_dend_height = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),cluster_rows = TRUE,row_title_gp = gpar(fontsize = 0),
           heatmap_legend_param =list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),title_gp = gpar(fontsize = 10),
                                      labels_gp = gpar(fontsize = 10)),
           name = "Flux",show_row_names = FALSE,row_names_gp=gpar(fontsize = 12),height  = unit(7, "cm"),width  = unit(15, "cm"))


pdf("/home/anoop/Desktop/CCHF/GEM/GroupLevel/FBA/Merged/Figures/personal/HeatMap.pdf",width = 12,height = 12)
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "right",merge_legend = TRUE)
dev.off()
```

```{r}
library(reshape)
R=read.delim("/home/anoop/Desktop/CCHF/GEM/Association/Selected.txt",header = TRUE)
head(R)
Cas=cast(R)
head(Cas)
write.table((Cas),file="/home/anoop/Desktop/CCHF/GEM/Association/Selected_Casted.txt",sep="\t",col.names = NA,quote = FALSE,na = "0")
```

```{r,eval=FALSE}
library(ComplexHeatmap)
library(circlize)
CelFr=read.delim("/home/anoop/Desktop/CCHF/GEM/Association/Selected_Casted2.txt",header = TRUE,check.names = FALSE,row.names = 1)

sample=read.delim("/home/anoop/Desktop/CCHF/GEM/Association/columnInfo2.txt",row.names = 1)

ha = HeatmapAnnotation(df = sample,show_annotation_name = FALSE,simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(SubSystem = list(direction = "horizontal",grid_width = unit(0.25, "cm"),grid_height = unit(0.25, "cm"),title_gp = gpar(fontsize = 10), 
                                                                   labels_gp = gpar(fontsize = 10))),
                       col = list(SubSystem=c("Transport reaction"="#9f8170","Amino acid transport"="#9fb6cd")))

col_corr = colorRamp2(c(1,0.95,0.9,0.75,0.5, 0,-0.5,-0.75,-0.9,-0.95,-1), 
                      c("#b20000","#cc0000","#e50000","#ff0000","#ff7f7f","grey","#7f7fff","#0000ff","#0000e5","#0000cc","#0000b2"))


info2=read.delim("/home/anoop/Desktop/CCHF/GEM/Association/rawInfo.txt",row.names = 1)
ha2 = rowAnnotation(df = info2,show_annotation_name = FALSE,simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0.25, "cm"),
                                                      grid_height = unit(0.25, "cm"),title_gp = gpar(fontsize = 10), 
                                                                   labels_gp = gpar(fontsize = 10)),
                       col = list(SubSystem=c("Glycolysis / Gluconeogenesis"="#94967b","Other"="#bfbfbf","Oxidative phosphorylation"="#eecff4","Pentose phosphate pathway"="#cecdff","Pyruvate metabolism"="#f2b894","TCA cycle"="#97f4e9")))

H1=Heatmap(CelFr,cluster_columns = TRUE,show_column_names = FALSE,col = col_corr,top_annotation = ha,row_dend_width = unit(0.5, "cm"),
           ,row_gap = unit(0.5, "mm"),left_annotation = ha2,row_split = info2$SubSystem,row_dend_reorder = TRUE,column_split = sample$SubSystem,
           column_dend_height = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),cluster_rows = TRUE,row_title_gp = gpar(fontsize = 0),
           heatmap_legend_param =list(grid_width = unit(0.25, "cm"),grid_height = unit(0.25, "cm"),title_gp = gpar(fontsize = 10),
                                      labels_gp = gpar(fontsize = 10)),
           name = "Spearman Rho",show_row_names = FALSE,row_names_gp=gpar(fontsize = 12),height  = unit(5, "cm"),width  = unit(8, "cm"))


pdf("/home/anoop/Desktop/CCHF/GEM/Association/HeatMap.pdf")
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "right",merge_legend = TRUE)
dev.off()
```

```{r,eval=FALSE}
library(ComplexHeatmap)
library(circlize)
CelFr=read.delim("/home/anoop/Desktop/CCHF/GEM/Association/EQN/Selected_Casted4.txt",header = TRUE,check.names = FALSE,row.names = 1)

sample=read.delim("/home/anoop/Desktop/CCHF/GEM/Association/EQN/columnInfo2.txt",row.names = 1)

ha = HeatmapAnnotation(df = sample,show_annotation_name = FALSE,simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(SubSystem = list(direction = "horizontal",grid_width = unit(0, "cm"),grid_height = unit(0, "cm"),title_gp = gpar(fontsize = 0), 
                                                                   labels_gp = gpar(fontsize = 0))),
                       col = list(SubSystem=c("Transport reaction"="#9f8170","Amino acid transport"="#9fb6cd")))

col_corr = colorRamp2(c(1,0.95,0.9,0.75,0.5, 0,-0.5,-0.75,-0.9,-0.95,-1), 
                      c("#b20000","#cc0000","#e50000","#ff0000","#ff7f7f","grey","#7f7fff","#0000ff","#0000e5","#0000cc","#0000b2"))


info2=read.delim("/home/anoop/Desktop/CCHF/GEM/Association/EQN/rawInfo.txt",row.names = 1)
ha2 = rowAnnotation(df = info2,show_annotation_name = FALSE,simple_anno_size = unit(0.2, "cm"),
                       annotation_legend_param = list(direction = "horizontal",grid_width = unit(0, "cm"),
                                                      grid_height = unit(0, "cm"),title_gp = gpar(fontsize = 0), 
                                                                   labels_gp = gpar(fontsize = 0)),
                       col = list(SubSystem=c("Glycolysis / Gluconeogenesis"="#94967b","Other"="#bfbfbf","Oxidative phosphorylation"="#eecff4","Pentose phosphate pathway"="#cecdff","Pyruvate metabolism"="#f2b894","TCA cycle"="#97f4e9")))

H1=Heatmap(CelFr,cluster_columns = TRUE,show_column_names = TRUE,col = col_corr,top_annotation = ha,row_dend_width = unit(0.5, "cm"),
           ,row_gap = unit(0.5, "mm"),left_annotation = ha2,row_split = info2$SubSystem,row_dend_reorder = TRUE,column_split = sample$SubSystem,column_names_gp = gpar(fontsize = 8),
           column_dend_height = unit(0.5, "cm"),column_title_gp =gpar(fontsize = 0),cluster_rows = TRUE,row_title_gp = gpar(fontsize = 0),
           heatmap_legend_param =list(grid_width = unit(0, "cm"),grid_height = unit(0, "cm"),title_gp = gpar(fontsize = 0),
                                      labels_gp = gpar(fontsize = 0)),
           name = "Spearman Rho",show_row_names = TRUE,row_names_gp=gpar(fontsize = 9),height  = unit(20, "cm"),width  = unit(40, "cm"))


pdf("/home/anoop/Desktop/CCHF/GEM/Association/EQN/HeatMap.pdf",width = 25,height = 18)
draw(H1,heatmap_legend_side = "right", annotation_legend_side = "left",merge_legend = TRUE)
dev.off()
```


```{r}


dd=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/Virus_Titre/FigIP.txt",header = TRUE)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/Virus_Titre/Fig.pdf")
ggplot(dd, aes(VirusTitre, TPM, colour = Module,fill=Module),size=2)+geom_smooth(alpha=0.2, method="lm",size=1.25)+
  xlab("log10(Virus Titre)")+
  geom_point(size=2,aes(colour = Module,fill=Module),pch=21)+
  scale_color_manual(values=c("c1"="#808497","c2"="#849780","c3"="#809793","c4"="#b2a47a"))+
  scale_fill_manual(values=c("c1"="#B7BDD8","c2"="#BDD8B7","c3"="#B7D8D2","c4"="#FFEBAF"))+
  theme(legend.title=element_text(size=10),legend.text=element_text(size=10),legend.key.size=unit(0.7,"line"),
        plot.title = element_text(hjust = 0.5,size =9),panel.grid.minor = element_blank(),legend.position = "bottom",
        axis.title.x=element_text(size=10),axis.title.y =element_text(size=10),
        axis.text.y=element_text(size=10),axis.text.x=element_text(size=10),
        plot.margin = margin(4,4,4,4, "cm"))
dev.off()

```

```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Heatmap/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.15, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.3, "cm"),
                                                              grid_height = unit(0.3, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Cohort=c("1. HC"="#4682B4","2. T0"="#a70000","3. T1"="#e4b2b2","4. T2"="#738957")))

Eu=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Heatmap/TPM_gene.txt",row.names = 1)
#col_fun1 = colorRamp2(c(1.5,0.75,0.5,0,-0.5,-0.75,-1.5), c("#ffff19","#99990f","#4c4c07","black","#003333","#00b2b2","#00ffff"))

col_fun1 = colorRamp2(c(1.5,0.75,0.5,0,-0.5,-0.75,-1.5), c("#ffff19","#99990f","#4c4c07","black","#203470","#3454b4","#4169e1"))

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Heatmap/GeneInfo.txt",row.names = 1)
gene_anno = rowAnnotation(df = gene, simple_anno_size = unit(0.15, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.3, "cm"),
                                                              grid_height = unit(0.3, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Cluster=c("c1"="#b7bdd8","c2"="#bdd8b7","c3"="#b7d8d2","c4"="#ffebaf")))


H1=Heatmap(as.matrix(t(scale(t(Eu)))),col=col_fun1,cluster_rows=TRUE,cluster_columns = FALSE,show_column_names = FALSE,
           na_col = "white",
           row_title_gp =gpar(fontsize = 0),row_split = gene$Cluster,left_annotation = gene_anno,
           row_dend_width = unit(0.75, "cm"),column_title_gp =gpar(fontsize = 0),top_annotation = chrt_anno,column_split = chrt$Cohort,
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(0.5, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.3, "cm"),grid_height = unit(0.3, "cm"),
                                       title_gp = gpar(fontsize = 8),
                                       labels_gp = gpar(fontsize = 8)),
           name = "Z-score(TPM)",show_row_names = TRUE,row_names_gp=gpar(fontsize =6),
           height  = unit(15, "cm"),width  = unit(7, "cm"))

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Heatmap/HeatmapIFN.pdf")
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r}
library(reshape)
R=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_CCM/Heatmap/COrr.txt",header = TRUE)
head(R)
Cas=cast(R)
head(Cas)
write.table((Cas),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_CCM/Heatmap/CorrCasted.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_CCM/Heatmap/CCM_Info6.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.25, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),na_col = "white",show_annotation_name = FALSE,
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 0),
                                                              labels_gp = gpar(fontsize = 10)),
                              col = list(CCM_in_cancer=c("CCM in cancer"="#5a5255"),
                                         TCA_cycle=c("TCA cycle"="#FFA500"),
                                         Electron_Transport_Chain=c("Electron Transport Chain (OXPHOS system in mitochondria)"="#559e83"),
                                         G_S_T_metabolism=c("Glycine, serine and threonine metabolism"="#ae5a41"),
                                         Glycolysis=c("Glycolysis"="#005900"),
                                         PPP=c("Pentose phosphate pathway"="#4682B4")))

Eu=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_CCM/Heatmap/CorrCasted.txt",row.names = 1)


col_fun1 = colorRamp2(c(1,0.5,0,-0.5,-1), c("#990000","#ff0000","white","#1919ff","#0000b2"))

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_CCM/Heatmap/Mod_Info.txt",row.names = 1)
gene_anno = rowAnnotation(df = gene, simple_anno_size = unit(0.25, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 10),
                                                              labels_gp = gpar(fontsize = 10)),
                              col = list(Module=c("c1"="#b7bdd8","c2"="#bdd8b7","c3"="#b7d8d2","c4"="#ffebaf")))


H1=Heatmap(as.matrix(Eu),col=col_fun1,cluster_rows=TRUE,cluster_columns = TRUE,show_column_names = FALSE,
           na_col = "grey",
           row_title_gp =gpar(fontsize = 0),row_split = gene$Module,left_annotation = gene_anno,
           row_dend_width = unit(0.75, "cm"),column_title_gp =gpar(fontsize = 0),top_annotation = chrt_anno,column_split = chrt$Cohort,
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(1, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 10),
                                       labels_gp = gpar(fontsize = 10)),
           name = "Spearman Rho",show_row_names = FALSE,row_names_gp=gpar(fontsize =6),
           height  = unit(15, "cm"),width  = unit(15, "cm"))

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_CCM/Heatmap/HeatmapCorr.pdf",width = 12,height = 12)
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```

```{r}
library(psych)
CCM=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/CCM.txt",
               check.names=FALSE,header = T)

IFN=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/IFN.txt",
               check.names=FALSE,header = T)

Res=corr.test(as.matrix(CCM),as.matrix(IFN),use = "pairwise",method="spearman",adjust="BH")

head(Res$r)
corr=melt(Res$r)
head(corr)
pval=melt(Res$p.adj)

corr$G1=pval$Var1
corr$G2=pval$Var2
corr$pvalue=pval$value

write.table(corr,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/CorrResults.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(Res$r,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/Rho.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(Res$p.adj,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/Padj.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/CCM_Gene5.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.25, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),na_col = "white",show_annotation_name = FALSE,
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 0),
                                                              labels_gp = gpar(fontsize = 10)),
                              col = list(TCA_cycle=c("TCA cycle"="#FFA500"),
                                         Oxphos=c("Electron Transport Chain (OXPHOS system in mitochondria)"="#559e83"),
                                         G_S_T=c("Glycine, serine and threonine metabolism"="#ae5a41"),
                                         Glycolysis=c("Glycolysis"="#005900"),
                                         PPP=c("Pentose phosphate pathway"="#4682B4")))


Eu=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/Rho.txt",row.names = 1)


col_fun1 = colorRamp2(c(1,0.75,0,-0.75,-1), c("#990000","#ff0000","white","#1919ff","#0000b2"))

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/IFN_Gene_Info.txt",row.names = 1)
gene_anno = rowAnnotation(df = gene, simple_anno_size = unit(0.25, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 10),
                                                              labels_gp = gpar(fontsize = 10)),
                              col = list(Module=c("c1"="#b7bdd8","c2"="#bdd8b7","c3"="#b7d8d2","c4"="#ffebaf")))


H1=Heatmap(as.matrix(Eu),col=col_fun1,cluster_rows=TRUE,cluster_columns = TRUE,show_column_names = FALSE,
           na_col = "grey",column_split = chrt$PPP,
           row_title_gp =gpar(fontsize = 0),row_split = gene$Module,left_annotation = gene_anno,top_annotation = chrt_anno,
           row_dend_width = unit(0.75, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(1, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 10),
                                       labels_gp = gpar(fontsize = 10)),
           name = "Spearman Rho",show_row_names = TRUE,row_names_gp=gpar(fontsize =8),
           height  = unit(20, "cm"),width  = unit(15, "cm"))

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HeatmapCorr_PPP.pdf",width = 12,height = 12)
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r}
library(psych)
library(reshape2)
CCM=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/CCM.txt",
               check.names=FALSE,header = T)

IFN=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/IFN.txt",
               check.names=FALSE,header = T)

Res=corr.test(as.matrix(CCM),as.matrix(IFN),use = "pairwise",method="spearman",adjust="BH")

head(Res$r)
corr=melt(Res$r)
head(corr)
pval=melt(Res$p.adj)

corr$G1=pval$Var1
corr$G2=pval$Var2
corr$pvalue=pval$value

write.table(corr,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/CorrResults.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(Res$r,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/Rho.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(Res$p.adj,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/Padj.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r}
library(psych)
CCM=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/CCM.txt",
               check.names=FALSE,header = T)

IFN=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/IFN.txt",
               check.names=FALSE,header = T)

Res=corr.test(as.matrix(CCM),as.matrix(IFN),use = "pairwise",method="spearman",adjust="BH")

head(Res$r)
corr=melt(Res$r)
head(corr)
pval=melt(Res$p.adj)

corr$G1=pval$Var1
corr$G2=pval$Var2
corr$pvalue=pval$value

write.table(corr,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/CorrResults.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(Res$r,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/Rho.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(Res$p.adj,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/Padj.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
library(psych)
CCM=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/CCM.txt",
               check.names=FALSE,header = T)

IFN=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/IFN.txt",
               check.names=FALSE,header = T)

Res=corr.test(as.matrix(CCM),as.matrix(IFN),use = "pairwise",method="spearman",adjust="BH")

head(Res$r)
corr=melt(Res$r)
head(corr)
pval=melt(Res$p.adj)

corr$G1=pval$Var1
corr$G2=pval$Var2
corr$pvalue=pval$value

write.table(corr,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/CorrResults.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(Res$r,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/Rho.txt",sep="\t",col.names = NA,quote = FALSE)

write.table(Res$p.adj,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/Padj.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
library(reshape)
R=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/Heatmap/SigniCorr.txt",header = TRUE)
head(R)
Cas=cast(R)
head(Cas)
write.table(t(Cas),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/Heatmap/SigniCorrCasted.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
library(reshape)
R=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/Heatmap/SigniCorr.txt",header = TRUE)
head(R)
Cas=cast(R)
head(Cas)
write.table(t(Cas),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/Heatmap/SigniCorrCasted.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
library(reshape)
R=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/Heatmap/SigniCorr.txt",header = TRUE)
head(R)
Cas=cast(R)
head(Cas)
write.table(t(Cas),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/Heatmap/SigniCorrCasted.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/Heatmap/CCM_Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.25, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),na_col = "white",show_annotation_name = FALSE,
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 0),
                                                              labels_gp = gpar(fontsize = 10)),
                              col = list(TCA_cycle=c("TCA cycle"="#FFA500"),
                                         OXPHOS=c("OXPHOS"="#559e83"),
                                         GST=c("Glycine, serine and threonine metabolism"="#ae5a41"),
                                         Glycolysis=c("Glycolysis"="#005900"),
                                         Pentose_phosphate_pathway=c("Pentose phosphate pathway"="#4682B4")))


Eu=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/Heatmap/SigniCorrCasted2.txt",row.names = 1)


col_fun1 = colorRamp2(c(1,0.75,0.5,0.25,0,-0.25,-0.5,-0.75,-1), c("#990000","#e50000","#ff4c4c","#ff9999","grey","#9999ff","#4c4cff","#0000ff","#0000cc"))

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/Heatmap/Module_Info.txt",row.names = 1)
gene_anno = rowAnnotation(df = gene, simple_anno_size = unit(0.25, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 10),
                                                              labels_gp = gpar(fontsize = 10)),
                              col = list(Module=c("cls1"="#b7bdd8","cls2"="#bdd8b7","cls3"="#b7d8d2","cls4"="#ffebaf")))


H1=Heatmap(as.matrix(Eu),col=col_fun1,cluster_rows=TRUE,cluster_columns = TRUE,show_column_names = FALSE,
           na_col = "grey",column_split = chrt$OXPHOS,
           row_title_gp =gpar(fontsize = 0),row_split = gene$Module,left_annotation = gene_anno,top_annotation = chrt_anno,
           row_dend_width = unit(0.75, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(1, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 10),
                                       labels_gp = gpar(fontsize = 10)),
           name = "Spearman Rho",show_row_names = TRUE,row_names_gp=gpar(fontsize =8),
           height  = unit(20, "cm"),width  = unit(15, "cm"))

pdf("//home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/HC/Heatmap/Heatmap_IFN_CCM_HC.pdf",width = 12,height = 12)
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```



```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/Heatmap/CCM_Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.25, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),na_col = "white",show_annotation_name = FALSE,
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 0),
                                                              labels_gp = gpar(fontsize = 10)),
                              col = list(TCA_cycle=c("TCA cycle"="#FFA500"),
                                         OXPHOS=c("OXPHOS"="#559e83"),
                                         GST=c("Glycine, serine and threonine metabolism"="#ae5a41"),
                                         Glycolysis=c("Glycolysis"="#005900"),
                                         Pentose_phosphate_pathway=c("Pentose phosphate pathway"="#4682B4")))


Eu=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/Heatmap/SigniCorrCasted2.txt",row.names = 1)


col_fun1 = colorRamp2(c(1,0.75,0.5,0.25,0,-0.25,-0.5,-0.75,-1), c("#990000","#e50000","#ff4c4c","#ff9999","grey","#9999ff","#4c4cff","#0000ff","#0000cc"))

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/Heatmap/Module_Info.txt",row.names = 1)
gene_anno = rowAnnotation(df = gene, simple_anno_size = unit(0.25, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 10),
                                                              labels_gp = gpar(fontsize = 10)),
                              col = list(Module=c("cls1"="#b7bdd8","cls2"="#bdd8b7","cls3"="#b7d8d2","cls4"="#ffebaf")))


H1=Heatmap(as.matrix(Eu),col=col_fun1,cluster_rows=TRUE,cluster_columns = TRUE,show_column_names = FALSE,
           na_col = "grey",column_split = chrt$OXPHOS,
           row_title_gp =gpar(fontsize = 0),row_split = gene$Module,left_annotation = gene_anno,top_annotation = chrt_anno,
           row_dend_width = unit(0.75, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(1, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 10),
                                       labels_gp = gpar(fontsize = 10)),
           name = "Spearman Rho",show_row_names = TRUE,row_names_gp=gpar(fontsize =8),
           height  = unit(20, "cm"),width  = unit(15, "cm"))

pdf("//home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T0/Heatmap/Heatmap_IFN_CCM_T0.pdf",width = 12,height = 12)
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/Heatmap/CCM_Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.25, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),na_col = "white",show_annotation_name = FALSE,
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 0),
                                                              labels_gp = gpar(fontsize = 10)),
                              col = list(TCA_cycle=c("TCA cycle"="#FFA500"),
                                         OXPHOS=c("OXPHOS"="#559e83"),
                                         GST=c("Glycine, serine and threonine metabolism"="#ae5a41"),
                                         Glycolysis=c("Glycolysis"="#005900"),
                                         Pentose_phosphate_pathway=c("Pentose phosphate pathway"="#4682B4")))


Eu=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/Heatmap/SigniCorrCasted2.txt",row.names = 1)


col_fun1 = colorRamp2(c(1,0.75,0.5,0.25,0,-0.25,-0.5,-0.75,-1), c("#990000","#e50000","#ff4c4c","#ff9999","grey","#9999ff","#4c4cff","#0000ff","#0000cc"))

gene=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/Heatmap/Module_Info.txt",row.names = 1)
gene_anno = rowAnnotation(df = gene, simple_anno_size = unit(0.25, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 10),
                                                              labels_gp = gpar(fontsize = 10)),
                              col = list(Module=c("cls1"="#b7bdd8","cls2"="#bdd8b7","cls3"="#b7d8d2","cls4"="#ffebaf")))


H1=Heatmap(as.matrix(Eu),col=col_fun1,cluster_rows=TRUE,cluster_columns = TRUE,show_column_names = FALSE,
           na_col = "grey",column_split = chrt$OXPHOS,
           row_title_gp =gpar(fontsize = 0),row_split = gene$Module,left_annotation = gene_anno,top_annotation = chrt_anno,
           row_dend_width = unit(0.75, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(1, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 10),
                                       labels_gp = gpar(fontsize = 10)),
           name = "Spearman Rho",show_row_names = TRUE,row_names_gp=gpar(fontsize =8),
           height  = unit(20, "cm"),width  = unit(15, "cm"))

pdf("//home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/DESeq2/Cluster/IFN_Genes/Corr_IFN_CCM/T2/Heatmap/Heatmap_IFN_CCM_T2.pdf",width = 12,height = 12)
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```


```{r,eval=FALSE}
library("tximport")
library("tximportData")
design=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Sample_Info.txt",header = TRUE)
files <- file.path("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq",design$Sample_ID, "CodingCount.txt")
names(files) <- design$Sample_ID
file.exists(files)
txi <- tximport(files, type="salmon",txIn=FALSE,geneIdCol="Name")
write.table(txi$abundance,file = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA_Check/TPM2.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r,eval=FALSE}
library(PCAtools)


data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA_Check/TPM2.txt",header=TRUE,row.names = 1,check.names = FALSE)

mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]

meta=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Sample_Info.txt",row.names = 1)
p <- PCAtools::pca((data), metadata = meta, removeVar = 0.1,scale=TRUE)
write.table(p$rotated,file = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA_Check/TPM_PCANoscale.txt",sep = "\t",col.names = NA,quote = FALSE)
head(p$variance)
?pca
```

```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA_Check/TPM_PCA2.txt",row.names = 1,check.names = FALSE)
gg <- merge(dat,aggregate(cbind(mean.pc1=PC1,mean.pc2=PC2)~Cohort,dat,mean),by="Cohort")
head(gg)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA_Check/TPM_PCATop.pdf")
ggplot(gg, aes(PC1,PC2,color=Cohort))+geom_point(size=3,aes(fill=Cohort,shape=SGS2))+
  scale_shape_manual(values=c("Healthy"=21, "Mild"=22, "Moderate"=23,"Severe"=24))+
  geom_point(aes(x=mean.pc1,y=mean.pc2,fill=Cohort,color=Cohort),size=3.5)+theme_gray()+
  geom_segment(aes(x=mean.pc1, y=mean.pc2, xend=PC1, yend=PC2))+
    scale_color_manual(values=c("Healthy"="#486129","T0"="#740000","T1"="#ca6666","T2"="#315b7d"))+
    scale_fill_manual(values=c("Healthy"="#738957","T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"))+
  labs(x="PC1, 28.79% variation",y="PC2, 6.7% variation")+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = Cohort), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```


```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA_Check/TPM_PCA.txt",row.names = 1,check.names = FALSE)
head(gg)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA_Check/TPM_PCA2.pdf")
ggplot(dat, aes(PC1,PC2,color=Cohort))+geom_point(size=3,aes(fill=Cohort,shape=SGS2))+
  scale_shape_manual(values=c("Healthy"=21, "Mild"=22, "Moderate"=23,"Severe"=24))+
    scale_color_manual(values=c("Healthy"="#486129","T0"="#740000","T1"="#ca6666","T2"="#315b7d"))+
    scale_fill_manual(values=c("Healthy"="#738957","T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"))+
  labs(x="PC1, 28.79% variation",y="PC2, 6.7% variation")+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = Cohort), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```


```{r}
library(psych)
CCM=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA_Check/data.txt",
               check.names=FALSE,header = T)

IFN=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA_Check/pc1.txt",
               check.names=FALSE,header = T)

Res=corr.test(as.matrix(CCM),as.matrix(IFN),use = "pairwise",method="spearman",adjust="BH")

head(Res$r)
corr=melt(Res$r)
head(corr)
pval=melt(Res$p.adj)

corr$G1=pval$Var1
corr$G2=pval$Var2
corr$pvalue=pval$value

write.table(corr,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA_Check/CorrResults.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r,eval=FALSE}
library(PCAtools)


data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM2.txt",header=TRUE,row.names = 1,check.names = FALSE)

mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]

meta=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/Sample_Info.txt",row.names = 1)
p <- PCAtools::pca((data), metadata = meta, removeVar = 0.1,scale = TRUE)
write.table(p$rotated,file = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_PCA_Scaled.txt",sep = "\t",col.names = NA,quote = FALSE)
head(p$variance)

```


```{r}


data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_PCA_Scaled.txt",row.names = 1,header = TRUE)

pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_PCA_Scaled.pdf")
ggplot(data, aes(PC1,PC2,color=SGS2))+geom_point(size=2,aes(fill=SGS2,color=SGS2,shape=Cohort))+
  scale_shape_manual(values=c("T0"=22, "T1"=23,"T2"=21))+theme_bw()+
    scale_color_manual(values=c("Mild"="#b29a50","Moderate"="#b28500","Severe"="#634a00"))+
    scale_fill_manual(values=c("Mild"="#ffdc73","Moderate"="#ffbf00","Severe"="#a67c00"))+
  labs(x="PC1, 98.7% variation",y="PC2, 0.4% variation")+
  geom_line(data=data,aes(group = pair,color=SGS2),arrow = arrow(type = "closed",ends = "last",length=unit(0.05, "inches")))+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = SGS2), alpha = 0.25,linetype = "dotted",level = 0.8)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```


```{r,eval=FALSE}
library(PCAtools)

data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2.txt",header=TRUE,row.names = 1,check.names = FALSE)

mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]

meta=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/Info.txt",row.names = 1)
p <- PCAtools::pca((data), metadata = meta, removeVar = 0.1,scale = TRUE)
write.table(p$rotated,file = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2_PCA_Top5000.txt",sep = "\t",col.names = NA,quote = FALSE)
head(p$variance)

```

```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2_PCA_Top5000.txt",row.names = 1,check.names = FALSE)
gg <- merge(dat,aggregate(cbind(mean.pc1=PC1,mean.pc2=PC2)~Cohort,dat,mean),by="Cohort")
head(gg)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2_PCA_Top5000.pdf")
ggplot(gg, aes(PC1,PC2,color=Cohort))+geom_point(size=3,aes(fill=Cohort),shape=21)+
  geom_point(aes(x=mean.pc1,y=mean.pc2,fill=Cohort,color=Cohort),size=3.5)+theme_gray()+
  geom_segment(aes(x=mean.pc1, y=mean.pc2, xend=PC1, yend=PC2))+
    scale_color_manual(values=c("Healthy"="#486129","T2"="#315b7d"))+
    scale_fill_manual(values=c("Healthy"="#738957","T2"="#4682B4"))+
  geom_label_repel(aes(label = rownames(dat)))+
  labs(x="PC1, 61.80% variation",y="PC2, 6.12% variation")+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = Cohort), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```


```{r,eval=FALSE}
library(umap)

data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2.txt",header=TRUE,row.names = 1,check.names = FALSE)
mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]

dataX=scale((data),center = TRUE, scale = TRUE)
head(dataX)
Art.umap = umap(t(dataX))
write.table(Art.umap$layout,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2_UMAP_Top5000.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2_UMAP_AllGenes.txt",row.names = 1,check.names = FALSE)
gg <- merge(dat,aggregate(cbind(mean.pc1=V1,mean.pc2=V2)~Cohort,dat,mean),by="Cohort")
head(gg)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TTPM_HC_T2_UMAP_Top5000.pdf")
ggplot(gg, aes(V1,V2,color=Cohort))+geom_point(size=3,aes(fill=Cohort),shape=21)+
  geom_point(aes(x=mean.pc1,y=mean.pc2,fill=Cohort,color=Cohort),size=3.5)+theme_gray()+
  geom_segment(aes(x=mean.pc1, y=mean.pc2, xend=V1, yend=V2))+
    scale_color_manual(values=c("Healthy"="#486129","T2"="#315b7d"))+
    scale_fill_manual(values=c("Healthy"="#738957","T2"="#4682B4"))+
  labs(x="UMAP1",y="UMAP2")+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = Cohort), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```

```{r,eval=FALSE}
library(umap)

data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/TPM.txt",header=TRUE,row.names = 1,check.names = FALSE)
mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]

dataX=scale((data),center = TRUE, scale = TRUE)

Art.umap = umap(t(dataX))
write.table(Art.umap$layout,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_All_Samples_UMAP_Top5000.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_All_Samples_UMAP_Top5000.txt",row.names = 1,check.names = FALSE)
gg <- merge(dat,aggregate(cbind(mean.pc1=V1,mean.pc2=V2)~Cohort,dat,mean),by="Cohort")
head(gg)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_All_Samples_UMAP_Top5000.pdf")
ggplot(gg, aes(V1,V2,color=Cohort))+geom_point(size=3,aes(fill=Cohort,shape=SGS2))+
  scale_shape_manual(values=c("Healthy"=21, "Mild"=22, "Moderate"=23,"Severe"=24))+
  geom_point(aes(x=mean.pc1,y=mean.pc2,fill=Cohort,color=Cohort),size=3.5)+theme_gray()+
  geom_segment(aes(x=mean.pc1, y=mean.pc2, xend=V1, yend=V2))+
    scale_color_manual(values=c("Healthy"="#486129","T0"="#740000","T1"="#ca6666","T2"="#315b7d"))+
    scale_fill_manual(values=c("Healthy"="#738957","T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"))+
  labs(x="UMAP1",y="UMAP2")+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = Cohort), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```


```{r,eval=FALSE}
library(PCAtools)

data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2_NoOutLier.txt",header=TRUE,row.names = 1,check.names = FALSE)

mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]

meta=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/Info2.txt",row.names = 1)
p <- PCAtools::pca((data), metadata = meta, removeVar = 0.1,scale = TRUE)
write.table(p$rotated,file = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/PCA_HC_T2_NoOutLier_Top5000.txt",sep = "\t",col.names = NA,quote = FALSE)
head(p$variance)

```

```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/PCA_HC_T2_NoOutLier_Top5000.txt",row.names = 1,check.names = FALSE)
gg <- merge(dat,aggregate(cbind(mean.pc1=PC1,mean.pc2=PC2)~Cohort,dat,mean),by="Cohort")
head(gg)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/PCA_HC_T2_NoOutLier_Top5000.pdf")
ggplot(gg, aes(PC1,PC2,color=Cohort))+geom_point(size=3,aes(fill=Cohort),shape=21)+
  geom_point(aes(x=mean.pc1,y=mean.pc2,fill=Cohort,color=Cohort),size=3.5)+theme_gray()+
  geom_segment(aes(x=mean.pc1, y=mean.pc2, xend=PC1, yend=PC2))+
    scale_color_manual(values=c("Healthy"="#486129","T2"="#315b7d"))+
    scale_fill_manual(values=c("Healthy"="#738957","T2"="#4682B4"))+
  geom_label_repel(aes(label = rownames(dat)))+
  labs(x="PC1, 61.53% variation",y="PC2, 6.25% variation")+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = Cohort), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```



```{r,eval=FALSE}
library(PCAtools)

data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2_NoOutLier.txt",header=TRUE,row.names = 1,check.names = FALSE)

meta=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/Info2.txt",row.names = 1)
p <- PCAtools::pca((data), metadata = meta, removeVar = 0.1,scale = TRUE)
write.table(p$rotated,file = "/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/PCA_Test_Outlier.txt",sep = "\t",col.names = NA,quote = FALSE)
head(p$variance)

```

```{r,fig.align = "center",fig.height=4, fig.width=5.5, echo=FALSE,fig.cap ="UMAP based sample distribution. Coloured by location"}
library(ggplot2)
library(ggrepel)
dat=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/PCA_Test.txt",row.names = 1,check.names = FALSE)
gg <- merge(dat,aggregate(cbind(mean.pc1=PC1,mean.pc2=PC2)~Cohort,dat,mean),by="Cohort")
head(gg)
pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/PCA_Test.pdf")
ggplot(gg, aes(PC1,PC2,color=Cohort))+geom_point(size=3,aes(fill=Cohort),shape=21)+
  geom_point(aes(x=mean.pc1,y=mean.pc2,fill=Cohort,color=Cohort),size=3.5)+theme_gray()+
  geom_segment(aes(x=mean.pc1, y=mean.pc2, xend=PC1, yend=PC2))+
    scale_color_manual(values=c("Healthy"="#486129","T2"="#315b7d"))+
    scale_fill_manual(values=c("Healthy"="#738957","T2"="#4682B4"))+
  labs(x="PC1, 28.67% variation",y="PC2, 6.02% variation")+
  geom_label_repel(aes(label = rownames(dat)))+
  theme(axis.title = element_text(size=10),legend.position = "right",plot.margin = margin(4,2,4,2, "cm"),
        legend.title=element_blank(),legend.text=element_text(size=10),legend.key.size = unit(0.5, "cm"))+
  stat_ellipse(geom = "polygon",aes(fill = Cohort), alpha = 0.25,linetype = "dotted",level = 0.9)+
  guides(shape = guide_legend(ncol =1,override.aes=list(fill="grey",color="grey")),color = guide_legend(ncol =1),fill = guide_legend(ncol =1))
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/TPM.txt",header=TRUE,row.names = 1,check.names = FALSE)
mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]
write.table(data,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Top5000_TPM.txt",sep="\t",col.names = NA,quote = FALSE)


dataX=scale((data),center = TRUE, scale = TRUE)
write.table(dataX,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Top5000_Scaled_TPM.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
library(reshape)
R=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Figure/Scaled_Corr.txt",header = TRUE)
head(R)
Cas=cast(R)
head(Cas)
write.table(t(Cas),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Figure/Scaled_CorrCasted.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Figure/SamInfo2.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Cohort=c("Healthy"="#738957","T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"),
                                         SGS=c("Healthy"="#738957","Mild"="#ffdc73","Moderate"="#ffbf00","Severe"="#a67c00")))

chrt_anno2 = rowAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Cohort=c("Healthy"="#738957","T0"="#a70000","T1"="#e4b2b2","T2"="#4682B4"),
                                         SGS=c("Healthy"="#738957","Mild"="#ffdc73","Moderate"="#ffbf00","Severe"="#a67c00")))


FBA=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Figure/Scaled_CorrCasted.txt",row.names = 1)


col_corr = colorRamp2(c(1,0.8,0.6,0.4, 0,-0.25,-0.5,-0.75,-1), c("#ff0000","#ff3232","#ff7f7f","#ffcccc","white","#e5e5ff",
"#7f7fff","#3232ff","#0000ff"))

col_corr = colorRamp2(c(1,0.95,0.9,0.85,0.8,0.75,0.7,0.65,0.6,0.55,0.5,0.45,0.4,0.2,0,-0.25,-0.5,-0.75,-1), c("#330000","#4c0000","#660000","#7f0000","#ff0000","#ff1919","#ff3232","#ff4c4c","#ff6666",
"#ffe5e5","#ffe5e5",                                                                                                  "#ffe5e5","#ffe5e5","#ffe5e5","white","#e5e5ff","#7f7fff","#3232ff","#0000ff"))

col_corr = colorRamp2(c(1,0.9,0.8, 0,-0.5,-0.7,-1), c("#7f0000","#b20000","white","white","white","#3232ff","#0000ff"))


H1=Heatmap(as.matrix((FBA)),col=col_corr,cluster_rows=TRUE,cluster_columns = TRUE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),top_annotation =  chrt_anno,left_annotation = chrt_anno2,
           row_dend_width = unit(1, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(1, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(1,0.7,0,-0.7,-1),
                                       labels_gp = gpar(fontsize = 8)),
           name = "spearman",show_row_names = FALSE,row_names_gp=gpar(fontsize = 6),
           height  = unit(11, "cm"),width  = unit(11, "cm"))


pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Figure/Scaled_CorrCasted.pdf",width = 10,height = 10)
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T0.txt",header=TRUE,row.names = 1,check.names = FALSE)
mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]

dataX=scale((data),center = TRUE, scale = TRUE)
write.table(dataX,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T0_5000_Scaled_TPM.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T1.txt",header=TRUE,row.names = 1,check.names = FALSE)
mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]

dataX=scale((data),center = TRUE, scale = TRUE)
write.table(dataX,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T1_5000_Scaled_TPM.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T2.txt",header=TRUE,row.names = 1,check.names = FALSE)
mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]

dataX=scale((data),center = TRUE, scale = TRUE)
write.table(dataX,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T2_5000_Scaled_TPM.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
library(tidyverse)  # data manipulation
library(cluster)    # clustering algorithms
library(factoextra)

tpm=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T0_5000_Scaled_TPM.txt",row.names = 1)
fviz_nbclust(tpm, kmeans, method = "silhouette")
```


```{r}
library(ConsensusClusterPlus)

tpm=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T0_5000_Scaled_TPM.txt",row.names = 1)
results = ConsensusClusterPlus(as.matrix(tpm),maxK=5,reps=50,pItem=0.8,pFeature=1,
                               title="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T0/",clusterAlg="km",
                               distance="euclidean",seed=123456,plot="png")

icl = calcICL(results,title="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T0/",plot="png")

write.table(as.data.frame(results[[2]][["consensusClass"]]),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T0/C2.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r}
library(tidyverse)  # data manipulation
library(cluster)    # clustering algorithms
library(factoextra)

tpm=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T1_5000_Scaled_TPM.txt",row.names = 1)
fviz_nbclust(tpm, kmeans, method = "silhouette")

```

```{r}
library(ConsensusClusterPlus)

tpm=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T1_5000_Scaled_TPM.txt",row.names = 1)
results = ConsensusClusterPlus(as.matrix(tpm),maxK=5,reps=50,pItem=0.8,pFeature=1,
                               title="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T1/",clusterAlg="km",
                               distance="euclidean",seed=123456,plot="png")

icl = calcICL(results,title="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T1/",plot="png")

write.table(as.data.frame(results[[3]][["consensusClass"]]),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T1/C3.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r}
library(ConsensusClusterPlus)

tpm=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T2_5000_Scaled_TPM.txt",row.names = 1)
results = ConsensusClusterPlus(as.matrix(tpm),maxK=5,reps=50,pItem=0.8,pFeature=1,
                               title="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T2/",clusterAlg="km",
                               distance="euclidean",seed=123456,plot="png")

icl = calcICL(results,title="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T2/",plot="png")

write.table(as.data.frame(results[[3]][["consensusClass"]]),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T2/C3.txt",sep="\t",col.names = NA,quote = FALSE)
```

```{r}
library(tidyverse)  # data manipulation
library(cluster)    # clustering algorithms
library(factoextra)

tpm=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T2_5000_Scaled_TPM.txt",row.names = 1)
fviz_nbclust(tpm, kmeans, method = "silhouette")

```

# KMeans


```{r}
library(tidyverse)  # data manipulation
library(cluster)    # clustering algorithms
library(factoextra)

data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T0_5000_Scaled_TPM.txt",header=TRUE,row.names = 1,check.names = FALSE)

df <- t(data)
distance <- get_dist(df)

fviz_dist(distance, gradient = list(low = "#00AFBB", mid = "white", high = "#FC4E07"),show_labels = FALSE)

fviz_nbclust(df, kmeans, method = "silhouette")
```

```{r}
set.seed(123456)
k2 <- kmeans(df, centers =6 , nstart = 50)
?fviz_cluster
fviz_cluster(k2, data = df)
write.table(as.data.frame(k2$cluster),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/Kmeans/T0.txt",sep = "\t",col.names = NA,quote = FALSE)
```

```{r}
library(tidyverse)  # data manipulation
library(cluster)    # clustering algorithms
library(factoextra)

data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T1_5000_Scaled_TPM.txt",header=TRUE,row.names = 1,check.names = FALSE)

df <- t(data)
distance <- get_dist(df)

fviz_dist(distance, gradient = list(low = "#00AFBB", mid = "white", high = "#FC4E07"),show_labels = FALSE)

fviz_nbclust(df, kmeans, method = "silhouette")
```
```{r}
set.seed(123456)
k2 <- kmeans(df, centers =2 , nstart = 50)

fviz_cluster(k2, data = df)
write.table(as.data.frame(k2$cluster),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/Kmeans/T1.txt",sep = "\t",col.names = NA,quote = FALSE)
```

```{r}
library(tidyverse)  # data manipulation
library(cluster)    # clustering algorithms
library(factoextra)

data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/T2_5000_Scaled_TPM.txt",header=TRUE,row.names = 1,check.names = FALSE)

df <- t(data)
distance <- get_dist(df)

fviz_dist(distance, gradient = list(low = "#00AFBB", mid = "white", high = "#FC4E07"),show_labels = FALSE)

fviz_nbclust(df, kmeans, method = "silhouette")
```

```{r}
set.seed(123456)
k2 <- kmeans(df, centers =2 , nstart = 50)

fviz_cluster(k2, data = df)
write.table(as.data.frame(k2$cluster),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/Severity/Kmeans/T2.txt",sep = "\t",col.names = NA,quote = FALSE)
```

# HC T2

```{r,eval=FALSE}
library(PCAtools)

data=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/PCA/TPM_HC_T2.txt",header=TRUE,row.names = 1,check.names = FALSE)

mads=apply(data,1,mad)
data=data[rev(order(mads))[1:5000],]

dataX=scale((data),center = TRUE, scale = TRUE)
write.table(dataX,file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/T2_HC/Scaled_TPM_5000.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r}
library(reshape)
R=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/T2_HC/Corr.txt",header = TRUE)
head(R)
Cas=cast(R)
head(Cas)
write.table(t(Cas),file="/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/T2_HC/CorrCasted.txt",sep="\t",col.names = NA,quote = FALSE)
```


```{r}
library(circlize)
library(ComplexHeatmap)

chrt=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/T2_HC/Info.txt",row.names = 1)

chrt_anno = HeatmapAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),annotation_name_side = "left",
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Cohort=c("Healthy"="#738957","T2"="#4682B4")))

chrt_anno2 = rowAnnotation(df = chrt, simple_anno_size = unit(0.2, "cm"),
                              annotation_name_gp = gpar(fontsize = 0),
                              annotation_legend_param  = list(grid_width = unit(0.5, "cm"),
                                                              grid_height = unit(0.5, "cm"),
                                                              title_gp = gpar(fontsize = 8),
                                                              labels_gp = gpar(fontsize = 8)),
                              col = list(Cohort=c("Healthy"="#738957","T2"="#4682B4")))


FBA=read.delim("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/T2_HC/CorrCasted.txt",row.names = 1)


col_corr = colorRamp2(c(1,0.8,0.6,0.4, 0,-0.25,-0.5,-0.75,-1), c("#ff0000","#ff3232","#ff7f7f","#ffcccc","white","#e5e5ff",
"#7f7fff","#3232ff","#0000ff"))

col_corr = colorRamp2(c(1,0.95,0.9,0.85,0.8,0.75,0.7,0.65,0.6,0.55,0.5,0.45,0.4,0.2,0,-0.25,-0.5,-0.75,-1), c("#330000","#4c0000","#660000","#7f0000","#ff0000","#ff1919","#ff3232","#ff4c4c","#ff6666",
"#ffe5e5","#ffe5e5",                                                                                                  "#ffe5e5","#ffe5e5","#ffe5e5","white","#e5e5ff","#7f7fff","#3232ff","#0000ff"))

col_corr = colorRamp2(c(1,0.9,0.8, 0,-0.5,-0.7,-1), c("#7f0000","#b20000","white","white","white","#3232ff","#0000ff"))

col_corr = colorRamp2(c(1,0.95,0.9,0.85,0.8,0.5, 0,-1), c("#660000","#7f0000","#ff4c4c","#7f5200","#ffa500","#ffd27f","white","#0000ff"))

col_corr = colorRamp2(c(1,0.95,0.85,0.5), c("#cc0000","#ff0000","#ffff00","white"))

H1=Heatmap(as.matrix((FBA)),col=col_corr,cluster_rows=TRUE,cluster_columns = TRUE,show_column_names = FALSE,na_col = "white",
           row_title_gp =gpar(fontsize = 0),top_annotation =  chrt_anno,left_annotation = chrt_anno2,
           row_dend_width = unit(1, "cm"),column_title_gp =gpar(fontsize = 0),
           column_names_gp =gpar(fontsize = 12),column_dend_height =unit(1, "cm"), 
           clustering_distance_columns = "euclidean",column_gap = unit(1, "mm"),
           heatmap_legend_param = list(grid_width = unit(0.5, "cm"),grid_height = unit(0.5, "cm"),
                                       title_gp = gpar(fontsize = 8), at = c(0,0.5,0.7,0.9,1),
                                       labels_gp = gpar(fontsize = 8)),
           name = "spearman",show_row_names = FALSE,row_names_gp=gpar(fontsize = 6),
           height  = unit(11, "cm"),width  = unit(11, "cm"))


pdf("/home/anoop/Desktop/CCHF/Laptop/CCHF/RNASeq/Re_Seq/Additioan_Rui/T2_HC/CorrCasted.pdf",width = 10,height = 10)
draw(H1, merge_legend = TRUE,heatmap_legend_side="right", annotation_legend_side="right")
dev.off()
```

# \textcolor{blue}{Reference}